---
title: 'DR-SC: installation and simulation'
author: "Wei Liu"
date: "`r Sys.Date()`"
output:
  pdf_document: default
  word_document: default
  html_document: default
vignette: |
  %\VignetteIndexEntry{Vignette Title} %\VignetteEngine{knitr::rmarkdown} %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Install the DR.SC
This vignette provides an introduction to the R package `DR.SC`, where the function `DR.SC` implements the model `DR-SC`, spatial clustering with hidden Markov random field using empirical Bayes.
The package can be installed with the command:

`library(remotes)`

`remotes::install_github("feiyoung/DR.SC")`

The package can be loaded with the command:
```{r}
library("DR.SC")
```

## Fit DR-SC using simulated data
### Generating the simulated data
First, we generate the spatial transcriptomics data with lattice neighborhood, i.e. ST platform.
```{r}
seu <- gendata_spatial(height=30, width=30,p=500, K=4)
head(seu@meta.data)
```

### Fitting model by using {DR.SC} function
For function DR.SC, users can specify the number of clusters $K$ or set {K=NULL} by using modified BIC(MBIC) to determine $K$. First, we try using user-specified number of clusters. Then we show the version chosen by MBIC.
```{r}
### Given K
library(Seurat)
seu <- NormalizeData(seu)
# choose 2000 variable features using Seurat
seu <- FindVariableFeatures(seu, nfeatures = 2000)
seu2 <- DR.SC(seu, K=4, platform = 'ST', verbose=F)
```

Using ARI to check the performance of clustering
```{r}
mclust::adjustedRandIndex(seu2$spatial.drsc.cluster, seu$true_clusters)
```

Show the spatial scatter plot for clusters
```{r}
spatialPlotClusters(seu2)
```

Show the tSNE plot based on the extracted features from DR-SC.
```{r}
drscPlot(seu2)
```
Show the UMAP plot based on the extracted features from DR-SC.
```{r}
drscPlot(seu2, visu.method = 'UMAP')
```

Use MBIC to choose number of clusters:
```{r}
seu2 <- DR.SC(seu, q=10, K=NULL, K_set =2:6, platform = 'ST', verbose=F)
mbicPlot(seu2)
```

## Visualizations of marker gene expression

### Ridge plots 
Visualize single cell expression distributions in each cluster from Seruat.
```{r}
genes <- c("gene-24","gene-68", "gene-95","gene-55")
RidgePlot(seu2, features = genes, ncol = 2)
```
### Violin plot
Visualize single cell expression distributions in each cluster
```{r}

VlnPlot(seu2, features = genes, ncol=2)
```
### Feature plot
We extract tSNE based on the features from DR-SC and then visualize feature expression in the low-dimensional space
 
```{r}
seu2 <- RunTSNE(seu2, reduction="dr-sc", reduction.key='drsc_tSNE_')
FeaturePlot(seu2, features = genes, reduction = 'tsne' ,ncol=2)

```


### Dot plots 
The size of the dot corresponds to the percentage of cells expressing the
feature in each cluster. The color represents the average expression level
```{r}
DotPlot(seu2, features = genes)
```

### Heatmap plot
Single cell heatmap of feature expression
```{r}
# standard scaling (no regression)
seu2 <- ScaleData(seu2)
DoHeatmap(subset(seu2, downsample = 500), features = genes, size = 5)
```