<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>DR-SC: simulation • DR.SC</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="DR-SC: simulation">
<meta property="og:description" content="DR.SC">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DR.SC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/DR-SC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/DR.SC.DLPFC.html">DR-SC: DLPFC Data Analysis</a>
    </li>
    <li>
      <a href="../articles/DR.SC.Simu.html">DR-SC: simulation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>DR-SC: simulation</h1>
                        <h4 data-toc-skip class="author">Wei Liu</h4>
            
            <h4 data-toc-skip class="date">2022-07-31</h4>
      
      
      <div class="hidden name"><code>DR.SC.Simu.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="generate-the-simulated-data">Generate the simulated data<a class="anchor" aria-label="anchor" href="#generate-the-simulated-data"></a>
</h2>
<p>First, we generate the spatial transcriptomics data with lattice neighborhood, i.e. ST platform by using the function <code>gendata_RNAExp</code> in <code>DR.SC</code> package, which is a Seurat object format. It is noted that the <code>meta.data</code> must include spatial coordinates in columns named “row” (x coordinates) and “col” (y coordinates)!</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DR.SC</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; Loading required package: spatstat.geom</span>
<span class="co">#&gt; Warning: package 'spatstat.geom' was built under R version 4.1.3</span>
<span class="co">#&gt; Loading required package: spatstat.data</span>
<span class="co">#&gt; Warning: package 'spatstat.data' was built under R version 4.1.3</span>
<span class="co">#&gt; spatstat.geom 2.4-0</span>
<span class="co">#&gt; DR.SC :  Joint dimension reduction and spatial clustering is conducted for</span>
<span class="co">#&gt; Single-cell RNA sequencing and spatial transcriptomics data, and more details can be referred to</span>
<span class="co">#&gt; Wei Liu, Xu Liao, Yi Yang, Huazhen Lin, Joe Yeong, Xiang Zhou, Xingjie Shi and Jin Liu. (2022) &lt;doi:10.1093/nar/gkac219&gt;. It is not only computationally efficient and scalable to the sample size increment, but also is capable of choosing the smoothness parameter and the number of clusters as well.</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/genspatialdata.html">gendata_RNAExp</a></span><span class="op">(</span>height<span class="op">=</span><span class="fl">30</span>, width<span class="op">=</span><span class="fl">30</span>,p<span class="op">=</span><span class="fl">500</span>, K<span class="op">=</span><span class="fl">4</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">seu</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span>
<span class="co">#&gt;          orig.ident nCount_RNA nFeature_RNA row col imagerow imagecol</span>
<span class="co">#&gt; spot1 SeuratProject    2363525          500   1   1        1        1</span>
<span class="co">#&gt; spot2 SeuratProject    2953614          500   2   1        2        1</span>
<span class="co">#&gt; spot3 SeuratProject    2595803          500   3   1        3        1</span>
<span class="co">#&gt; spot4 SeuratProject    4222306          500   4   1        4        1</span>
<span class="co">#&gt; spot5 SeuratProject    3861722          500   5   1        5        1</span>
<span class="co">#&gt; spot6 SeuratProject    5614859          500   6   1        6        1</span>
<span class="co">#&gt;       true_clusters</span>
<span class="co">#&gt; spot1             1</span>
<span class="co">#&gt; spot2             1</span>
<span class="co">#&gt; spot3             1</span>
<span class="co">#&gt; spot4             4</span>
<span class="co">#&gt; spot5             4</span>
<span class="co">#&gt; spot6             4</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="fit-dr-sc-using-simulated-data">Fit DR-SC using simulated data<a class="anchor" aria-label="anchor" href="#fit-dr-sc-using-simulated-data"></a>
</h2>
<div class="section level3">
<h3 id="data-preprocessing">Data preprocessing<a class="anchor" aria-label="anchor" href="#data-preprocessing"></a>
</h3>
<p>This preprocessing includes Log-normalization and feature selection. Here we select highly variable genes for example first. The selected genes’ names are saved in “<a href="mailto:seu@assays%24RNA@var.features" class="email">seu@assays$RNA@var.features</a>”</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### Given K</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: package 'Seurat' was built under R version 4.1.3</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="co">#&gt; Attaching sp</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">seu</span><span class="op">)</span>
<span class="co"># choose highly variable features using Seurat</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">seu</span>, nfeatures <span class="op">=</span> <span class="fl">400</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fit-dr-sc-based-on-highly-variable-geneshvgs">Fit DR-SC based on highly variable genes(HVGs)<a class="anchor" aria-label="anchor" href="#fit-dr-sc-based-on-highly-variable-geneshvgs"></a>
</h3>
<p>For function <code>DR.SC</code>, users can specify the number of clusters <span class="math inline">\(K\)</span> or set <code>K</code> to be an integer vector by using modified BIC(MBIC) to determine <span class="math inline">\(K\)</span>. First, we try using user-specified number of clusters. Then we show the version chosen by MBIC.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### Given K</span>
<span class="va">seu2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DR.SC.html">DR.SC</a></span><span class="op">(</span><span class="va">seu</span>, K<span class="op">=</span><span class="fl">4</span>, platform <span class="op">=</span> <span class="st">'ST'</span>,  verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="co">#&gt; Neighbors were identified for 900 out of 900 spots.</span>
<span class="co">#&gt; Fit DR-SC model...</span>
<span class="co">#&gt; Finish DR-SC model fitting</span></code></pre></div>
<p>After finishing model fitting, we use ajusted rand index (<code>ARI</code>) to check the performance of clustering</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">mclust</span><span class="fu">::</span><span class="fu"><a href="https://mclust-org.github.io/mclust/reference/adjustedRandIndex.html" class="external-link">adjustedRandIndex</a></span><span class="op">(</span><span class="va">seu2</span><span class="op">$</span><span class="va">spatial.drsc.cluster</span>, <span class="va">seu</span><span class="op">$</span><span class="va">true_clusters</span><span class="op">)</span>
<span class="co">#&gt; [1] 1</span></code></pre></div>
<p>Next, we show the application of DR-SC in visualization. First, we can visualize the clusters from DR-SC on the spatial coordinates.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/spatialPlotClusters.html">spatialPlotClusters</a></span><span class="op">(</span><span class="va">seu2</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>We can also visualize the clusters from DR-SC on the two-dimensional tSNE based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/drscPlot.html">drscPlot</a></span><span class="op">(</span><span class="va">seu2</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-7-1.png" width="700"> Show the UMAP plot based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/drscPlot.html">drscPlot</a></span><span class="op">(</span><span class="va">seu2</span>, visu.method <span class="op">=</span> <span class="st">'UMAP'</span><span class="op">)</span>
<span class="co">#&gt; Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric</span>
<span class="co">#&gt; To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'</span>
<span class="co">#&gt; This message will be shown once per session</span>
<span class="co">#&gt; Warning: Invalid name supplied, making object name syntactically valid. New</span>
<span class="co">#&gt; object name is RunUMAP.RNA.dr.sc; see ?make.names for more details on syntax</span>
<span class="co">#&gt; validity</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>Use MBIC to choose number of clusters:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seu2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DR.SC.html">DR.SC</a></span><span class="op">(</span><span class="va">seu</span>, q<span class="op">=</span><span class="fl">10</span>, K<span class="op">=</span><span class="fl">2</span><span class="op">:</span><span class="fl">6</span>, platform <span class="op">=</span> <span class="st">'ST'</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="co">#&gt; Neighbors were identified for 900 out of 900 spots.</span>
<span class="co">#&gt; Fit DR-SC model...</span>
<span class="co">#&gt; Starting parallel computing intial values...</span>
<span class="co">#&gt; Finish DR-SC model fitting</span>
<span class="fu"><a href="../reference/mbicPlot.html">mbicPlot</a></span><span class="op">(</span><span class="va">seu2</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="fit-dr-sc-based-on-spatially-variable-genessvgs">Fit DR-SC based on spatially variable genes(SVGs)<a class="anchor" aria-label="anchor" href="#fit-dr-sc-based-on-spatially-variable-genessvgs"></a>
</h3>
<p>First, we select the spatilly variable genes using funciton <code>FindSVGs</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### Given K</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">seu</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="co"># choose 400 spatially variable features using FindSVGs</span>
<span class="va">seus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindSVGs.html">FindSVGs</a></span><span class="op">(</span><span class="va">seu</span>, nfeatures <span class="op">=</span> <span class="fl">400</span>, verbose <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="va">seu2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DR.SC.html">DR.SC</a></span><span class="op">(</span><span class="va">seus</span>, K<span class="op">=</span><span class="fl">4</span>, platform <span class="op">=</span> <span class="st">'ST'</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="co">#&gt; Neighbors were identified for 900 out of 900 spots.</span>
<span class="co">#&gt; Fit DR-SC model...</span>
<span class="co">#&gt; Finish DR-SC model fitting</span></code></pre></div>
<p>Using ARI to check the performance of clustering</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">mclust</span><span class="fu">::</span><span class="fu"><a href="https://mclust-org.github.io/mclust/reference/adjustedRandIndex.html" class="external-link">adjustedRandIndex</a></span><span class="op">(</span><span class="va">seu2</span><span class="op">$</span><span class="va">spatial.drsc.cluster</span>, <span class="va">seu</span><span class="op">$</span><span class="va">true_clusters</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.9971493</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="dr-sc-can-enhance-visualization">DR-SC can enhance visualization<a class="anchor" aria-label="anchor" href="#dr-sc-can-enhance-visualization"></a>
</h3>
<p>Show the spatial scatter plot for clusters</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/spatialPlotClusters.html">spatialPlotClusters</a></span><span class="op">(</span><span class="va">seu2</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>Show the tSNE plot based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/drscPlot.html">drscPlot</a></span><span class="op">(</span><span class="va">seu2</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-13-1.png" width="700"> Show the UMAP plot based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/drscPlot.html">drscPlot</a></span><span class="op">(</span><span class="va">seu2</span>, visu.method <span class="op">=</span> <span class="st">'UMAP'</span><span class="op">)</span>
<span class="co">#&gt; Warning: Invalid name supplied, making object name syntactically valid. New</span>
<span class="co">#&gt; object name is RunUMAP.RNA.dr.sc; see ?make.names for more details on syntax</span>
<span class="co">#&gt; validity</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="dr-sc-can-automatically-determine-the-number-of-clusters">DR-SC can automatically determine the number of clusters<a class="anchor" aria-label="anchor" href="#dr-sc-can-automatically-determine-the-number-of-clusters"></a>
</h3>
<p>Use MBIC to choose number of clusters:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seu2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DR.SC.html">DR.SC</a></span><span class="op">(</span><span class="va">seus</span>, q<span class="op">=</span><span class="fl">10</span>, K<span class="op">=</span><span class="fl">2</span><span class="op">:</span><span class="fl">6</span>, platform <span class="op">=</span> <span class="st">'ST'</span>,  verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="co">#&gt; Neighbors were identified for 900 out of 900 spots.</span>
<span class="co">#&gt; Fit DR-SC model...</span>
<span class="co">#&gt; Starting parallel computing intial values...</span>
<span class="co">#&gt; Finish DR-SC model fitting</span>
<span class="fu"><a href="../reference/mbicPlot.html">mbicPlot</a></span><span class="op">(</span><span class="va">seu2</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># or plot BIC or AIC</span>
<span class="co"># mbicPlot(seu2, criteria = 'BIC')</span>
<span class="co"># mbicPlot(seu2, criteria = 'AIC')</span>
<span class="co"># tune pen.const</span>
<span class="va">seu2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectModel.html">selectModel</a></span><span class="op">(</span><span class="va">seu2</span>, pen.const <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span>
<span class="fu"><a href="../reference/mbicPlot.html">mbicPlot</a></span><span class="op">(</span><span class="va">seu2</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-15-2.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="dr-sc-can-help-differentially-expression-analysis">DR-SC can help differentially expression analysis<a class="anchor" aria-label="anchor" href="#dr-sc-can-help-differentially-expression-analysis"></a>
</h2>
<p>Conduct visualization of marker gene expression. ### Ridge plots Visualize single cell expression distributions in each cluster from Seruat.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html" class="external-link">FindAllMarkers</a></span><span class="op">(</span><span class="va">seu2</span><span class="op">)</span>
<span class="co">#&gt; Calculating cluster cluster1</span>
<span class="co">#&gt; Calculating cluster cluster2</span>
<span class="co">#&gt; Calculating cluster cluster3</span>
<span class="co">#&gt; Calculating cluster cluster4</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span> <span class="op">)</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.1.3</span>
<span class="co"># Find the top 1 marker genes, user can change n to access more marker genes</span>
<span class="va">dat</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_n</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">1</span>, wt <span class="op">=</span> <span class="va">avg_log2FC</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">top</span>
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">top</span><span class="op">$</span><span class="va">gene</span>
<span class="fu"><a href="https://satijalab.org/seurat/reference/RidgePlot.html" class="external-link">RidgePlot</a></span><span class="op">(</span><span class="va">seu2</span>, features <span class="op">=</span> <span class="va">genes</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="co">#&gt; Picking joint bandwidth of 0.297</span>
<span class="co">#&gt; Picking joint bandwidth of 0.242</span>
<span class="co">#&gt; Picking joint bandwidth of 0.335</span>
<span class="co">#&gt; Picking joint bandwidth of 0.322</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-16-1.png" width="700"> ### Violin plot Visualize single cell expression distributions in each cluster</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://satijalab.org/seurat/reference/VlnPlot.html" class="external-link">VlnPlot</a></span><span class="op">(</span><span class="va">seu2</span>, features <span class="op">=</span> <span class="va">genes</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-17-1.png" width="700"> ### Feature plot We extract tSNE based on the features from DR-SC and then visualize feature expression in the low-dimensional space</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seu2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunTSNE.html" class="external-link">RunTSNE</a></span><span class="op">(</span><span class="va">seu2</span>, reduction<span class="op">=</span><span class="st">"dr-sc"</span>, reduction.key<span class="op">=</span><span class="st">'drsctSNE_'</span><span class="op">)</span>
<span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">seu2</span>, features <span class="op">=</span> <span class="va">genes</span>, reduction <span class="op">=</span> <span class="st">'tsne'</span> ,ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-18-1.png" width="700"></p>
<div class="section level3">
<h3 id="dot-plots">Dot plots<a class="anchor" aria-label="anchor" href="#dot-plots"></a>
</h3>
<p>The size of the dot corresponds to the percentage of cells expressing the feature in each cluster. The color represents the average expression level</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://satijalab.org/seurat/reference/DotPlot.html" class="external-link">DotPlot</a></span><span class="op">(</span><span class="va">seu2</span>, features <span class="op">=</span> <span class="va">genes</span><span class="op">)</span>
<span class="co">#&gt; Warning: Scaling data with a low number of groups may produce misleading results</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="heatmap-plot">Heatmap plot<a class="anchor" aria-label="anchor" href="#heatmap-plot"></a>
</h3>
<p>Single cell heatmap of feature expression</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># standard scaling (no regression)</span>
<span class="va">dat</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_n</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">30</span>, wt <span class="op">=</span> <span class="va">avg_log2FC</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">top</span>
<span class="co">### select the marker genes that are also the variable genes.</span>

<span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">top</span><span class="op">$</span><span class="va">gene</span>, <span class="va">seu2</span><span class="op">[[</span><span class="st">'RNA'</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">var.features</span><span class="op">)</span>
<span class="co">## Change the HVGs to SVGs</span>
<span class="co">#  &lt;- topSVGs(seu2, 400)</span>
<span class="va">seu2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">seu2</span>, verbose <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="fu"><a href="https://satijalab.org/seurat/reference/DoHeatmap.html" class="external-link">DoHeatmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">seu2</span>, downsample <span class="op">=</span> <span class="fl">500</span><span class="op">)</span>,features <span class="op">=</span> <span class="va">genes</span>, size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span>
<span class="co">#&gt; Running under: Windows 10 x64 (build 22000)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] LC_COLLATE=English_Singapore.1252  LC_CTYPE=English_Singapore.1252   </span>
<span class="co">#&gt; [3] LC_MONETARY=English_Singapore.1252 LC_NUMERIC=C                      </span>
<span class="co">#&gt; [5] LC_TIME=English_Singapore.1252    </span>
<span class="co">#&gt; system code page: 936</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] dplyr_1.0.9         sp_1.5-0            SeuratObject_4.1.0 </span>
<span class="co">#&gt; [4] Seurat_4.1.1        DR.SC_2.9           spatstat.geom_2.4-0</span>
<span class="co">#&gt; [7] spatstat.data_2.2-0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] GiRaF_1.0.1           systemfonts_1.0.4     plyr_1.8.7           </span>
<span class="co">#&gt;   [4] igraph_1.3.2          lazyeval_0.2.2        splines_4.1.2        </span>
<span class="co">#&gt;   [7] listenv_0.8.0         scattermore_0.8       ggplot2_3.3.6        </span>
<span class="co">#&gt;  [10] digest_0.6.29         htmltools_0.5.2       fansi_1.0.3          </span>
<span class="co">#&gt;  [13] magrittr_2.0.3        memoise_2.0.1         tensor_1.5           </span>
<span class="co">#&gt;  [16] cluster_2.1.2         ROCR_1.0-11           limma_3.50.1         </span>
<span class="co">#&gt;  [19] globals_0.15.0        matrixStats_0.62.0    pkgdown_2.0.2        </span>
<span class="co">#&gt;  [22] spatstat.sparse_2.1-1 colorspace_2.0-3      ggrepel_0.9.1        </span>
<span class="co">#&gt;  [25] textshaping_0.3.6     xfun_0.29             crayon_1.5.1         </span>
<span class="co">#&gt;  [28] jsonlite_1.8.0        progressr_0.10.1      survival_3.2-13      </span>
<span class="co">#&gt;  [31] zoo_1.8-10            glue_1.6.2            polyclip_1.10-0      </span>
<span class="co">#&gt;  [34] gtable_0.3.0          leiden_0.4.2          future.apply_1.9.0   </span>
<span class="co">#&gt;  [37] BiocGenerics_0.40.0   abind_1.4-5           scales_1.2.0         </span>
<span class="co">#&gt;  [40] DBI_1.1.2             spatstat.random_2.2-0 miniUI_0.1.1.1       </span>
<span class="co">#&gt;  [43] Rcpp_1.0.8.3          viridisLite_0.4.0     xtable_1.8-4         </span>
<span class="co">#&gt;  [46] reticulate_1.25       spatstat.core_2.4-4   mclust_5.4.10        </span>
<span class="co">#&gt;  [49] stats4_4.1.2          htmlwidgets_1.5.4     httr_1.4.3           </span>
<span class="co">#&gt;  [52] RColorBrewer_1.1-3    ellipsis_0.3.2        ica_1.0-2            </span>
<span class="co">#&gt;  [55] pkgconfig_2.0.3       farver_2.1.0          sass_0.4.1           </span>
<span class="co">#&gt;  [58] uwot_0.1.11           deldir_1.0-6          utf8_1.2.2           </span>
<span class="co">#&gt;  [61] tidyselect_1.1.2      labeling_0.4.2        rlang_1.0.2          </span>
<span class="co">#&gt;  [64] reshape2_1.4.4        later_1.3.0           munsell_0.5.0        </span>
<span class="co">#&gt;  [67] tools_4.1.2           cachem_1.0.6          cli_3.2.0            </span>
<span class="co">#&gt;  [70] generics_0.1.2        ggridges_0.5.3        evaluate_0.15        </span>
<span class="co">#&gt;  [73] stringr_1.4.0         fastmap_1.1.0         yaml_2.3.5           </span>
<span class="co">#&gt;  [76] ragg_1.2.2            goftest_1.2-3         knitr_1.37           </span>
<span class="co">#&gt;  [79] fs_1.5.2              fitdistrplus_1.1-8    purrr_0.3.4          </span>
<span class="co">#&gt;  [82] RANN_2.6.1            pbapply_1.5-0         future_1.26.1        </span>
<span class="co">#&gt;  [85] nlme_3.1-155          mime_0.12             ggrastr_1.0.1        </span>
<span class="co">#&gt;  [88] compiler_4.1.2        rstudioapi_0.13       beeswarm_0.4.0       </span>
<span class="co">#&gt;  [91] plotly_4.10.0         png_0.1-7             spatstat.utils_2.3-1 </span>
<span class="co">#&gt;  [94] tibble_3.1.7          bslib_0.3.1           stringi_1.7.6        </span>
<span class="co">#&gt;  [97] highr_0.9             desc_1.4.0            rgeos_0.5-9          </span>
<span class="co">#&gt; [100] lattice_0.20-45       Matrix_1.4-0          vctrs_0.4.1          </span>
<span class="co">#&gt; [103] CompQuadForm_1.4.3    pillar_1.7.0          lifecycle_1.0.1      </span>
<span class="co">#&gt; [106] lmtest_0.9-40         jquerylib_0.1.4       RcppAnnoy_0.0.19     </span>
<span class="co">#&gt; [109] data.table_1.14.2     cowplot_1.1.1         irlba_2.3.5          </span>
<span class="co">#&gt; [112] httpuv_1.6.5          patchwork_1.1.1       R6_2.5.1             </span>
<span class="co">#&gt; [115] promises_1.2.0.1      KernSmooth_2.23-20    gridExtra_2.3        </span>
<span class="co">#&gt; [118] vipor_0.4.5           parallelly_1.32.0     codetools_0.2-18     </span>
<span class="co">#&gt; [121] MASS_7.3-55           assertthat_0.2.1      rprojroot_2.0.3      </span>
<span class="co">#&gt; [124] withr_2.5.0           sctransform_0.3.3     S4Vectors_0.32.3     </span>
<span class="co">#&gt; [127] mgcv_1.8-39           grid_4.1.2            rpart_4.1.16         </span>
<span class="co">#&gt; [130] tidyr_1.2.0           rmarkdown_2.11        Rtsne_0.16           </span>
<span class="co">#&gt; [133] shiny_1.7.1           ggbeeswarm_0.6.0</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Wei Liu, Yi Yang, Jin Liu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
