<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>DR-SC: DLPFC Data Analysis • DR.SC</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="DR-SC: DLPFC Data Analysis">
<meta property="og:description" content="DR.SC">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DR.SC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/DR-SC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/DR.SC.DLPFC.html">DR-SC: DLPFC Data Analysis</a>
    </li>
    <li>
      <a href="../articles/DR.SC.Simu.html">DR-SC: simulation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>DR-SC: DLPFC Data Analysis</h1>
                        <h4 data-toc-skip class="author">Wei Liu</h4>
            
            <h4 data-toc-skip class="date">2022-07-31</h4>
      
      
      <div class="hidden name"><code>DR.SC.DLPFC.Rmd</code></div>

    </div>

    
    
<p>The package can be loaded with the command:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"DR.SC"</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; Loading required package: spatstat.geom</span>
<span class="co">#&gt; Warning: package 'spatstat.geom' was built under R version 4.1.3</span>
<span class="co">#&gt; Loading required package: spatstat.data</span>
<span class="co">#&gt; Warning: package 'spatstat.data' was built under R version 4.1.3</span>
<span class="co">#&gt; spatstat.geom 2.4-0</span>
<span class="co">#&gt; DR.SC :  Joint dimension reduction and spatial clustering is conducted for</span>
<span class="co">#&gt; Single-cell RNA sequencing and spatial transcriptomics data, and more details can be referred to</span>
<span class="co">#&gt; Wei Liu, Xu Liao, Yi Yang, Huazhen Lin, Joe Yeong, Xiang Zhou, Xingjie Shi and Jin Liu. (2022) &lt;doi:10.1093/nar/gkac219&gt;. It is not only computationally efficient and scalable to the sample size increment, but also is capable of choosing the smoothness parameter and the number of clusters as well.</span></code></pre></div>
<div class="section level2">
<h2 id="fit-dr-sc-using-real-data-dlpfc">Fit DR-SC using real data DLPFC<a class="anchor" aria-label="anchor" href="#fit-dr-sc-using-real-data-dlpfc"></a>
</h2>
<p>To fit DR-SC model, we first require to create a Seurat object with meta.data including spatial coordinates in columns must named “row” (x coordinates) and “col” (y coordinates)!</p>
<div class="section level3">
<h3 id="prepare-seurat-object-for-dr-sc">Prepare Seurat object for DR-SC<a class="anchor" aria-label="anchor" href="#prepare-seurat-object-for-dr-sc"></a>
</h3>
<p>We start this tutorial with creating the Seurat object. Users who are familar with Seurat object, can skip this subsection and go to next subsection.</p>
<p>First, we load the subset data of DLPFC 151510, dlpfc151510. Although it is saved as a Seurat object prepared for DR-SC, we re-create the Seurat object to show how to prepare the Seurat object for DR-SC.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"dlpfc151510"</span>, package <span class="op">=</span> <span class="st">'DR.SC'</span><span class="op">)</span></code></pre></div>
<p>Second, we create a Seurat object using the count matrix in dlpfc151510 and the spatial coordinates in the meta.data slot.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span>
<span class="va">count</span> <span class="op">&lt;-</span> <span class="va">dlpfc151510</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span>
<span class="va">meta_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>row<span class="op">=</span><span class="va">dlpfc151510</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">row</span>, col<span class="op">=</span><span class="va">dlpfc151510</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">col</span>, annotation<span class="op">=</span><span class="va">dlpfc151510</span><span class="op">$</span><span class="va">annotation</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">meta_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">count</span><span class="op">)</span>
<span class="co">## create Seurat object</span>
<span class="va">dlpfc151510</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts<span class="op">=</span><span class="va">count</span>, meta.data <span class="op">=</span> <span class="va">meta_data</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">dlpfc151510</span><span class="op">)</span>
<span class="co">#&gt;                       orig.ident nCount_RNA nFeature_RNA row col annotation</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1 SeuratProject        879          223  50 102     Layer1</span>
<span class="co">#&gt; AAACACCAATAACTGC-1 SeuratProject        887          251  59  19     Layer5</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1 SeuratProject       1109          253  14  94     Layer3</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1 SeuratProject       1652          325  43   9     Layer4</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1 SeuratProject       1389          289  47  13     Layer4</span>
<span class="co">#&gt; AAACAGTGTTCCTGGG-1 SeuratProject        758          226  73  43     Layer6</span>
<span class="co">#&gt; AAACATGGTGAGAGGA-1 SeuratProject        750          235  62   0         WM</span>
<span class="co">#&gt; AAACATTTCCCGGATT-1 SeuratProject       1946          332  61  97     Layer3</span>
<span class="co">#&gt; AAACCCGAACGAAATC-1 SeuratProject        798          244  45 115     Layer1</span>
<span class="co">#&gt; AAACCGGAAATGTTAA-1 SeuratProject        796          217  54 124     Layer2</span></code></pre></div>
<p>Until now, the data preparation with Seurat object format is finished, and we can go to next step: preprocessing.</p>
</div>
<div class="section level3">
<h3 id="data-preprocessing">Data preprocessing<a class="anchor" aria-label="anchor" href="#data-preprocessing"></a>
</h3>
<p>This preprocessing includes Log-normalization and feature selection. Here we select highly variable genes for example first. The selected genes’ names are saved in “<a href="mailto:seu@assays%24RNA@var.features" class="email">seu@assays$RNA@var.features</a>”</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># standard log-normalization</span>
<span class="va">dlpfc151510</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">dlpfc151510</span>, verbose <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="co"># choose 500 highly variable features</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">dlpfc151510</span>, nfeatures <span class="op">=</span> <span class="fl">500</span>, verbose <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="va">seu</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">var.features</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>
<span class="co">#&gt;  [1] "ENSG00000123560" "ENSG00000197971" "ENSG00000122585" "ENSG00000211592"</span>
<span class="co">#&gt;  [5] "ENSG00000244734" "ENSG00000110484" "ENSG00000091513" "ENSG00000109846"</span>
<span class="co">#&gt;  [9] "ENSG00000173786" "ENSG00000131095"</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="fit-dr-sc-model-using-500-highly-variable-features">Fit DR-SC model using 500 highly variable features<a class="anchor" aria-label="anchor" href="#fit-dr-sc-model-using-500-highly-variable-features"></a>
</h2>
<p>We fit the DR-SC model by using the highly variable genes.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### Given K</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DR.SC.html">DR.SC</a></span><span class="op">(</span><span class="va">seu</span>, K<span class="op">=</span><span class="fl">7</span>, platform <span class="op">=</span> <span class="st">'Visium'</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="co">#&gt; Neighbors were identified for 4634 out of 4634 spots.</span>
<span class="co">#&gt; Fit DR-SC model...</span>
<span class="co">#&gt; Finish DR-SC model fitting</span></code></pre></div>
<div class="section level3">
<h3 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/spatialPlotClusters.html">spatialPlotClusters</a></span><span class="op">(</span><span class="va">seu</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.DLPFC_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>Show the tSNE plot based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/drscPlot.html">drscPlot</a></span><span class="op">(</span><span class="va">seu</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.DLPFC_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>Show the UMAP plot based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/drscPlot.html">drscPlot</a></span><span class="op">(</span><span class="va">seu</span>, visu.method <span class="op">=</span> <span class="st">'UMAP'</span><span class="op">)</span>
<span class="co">#&gt; Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric</span>
<span class="co">#&gt; To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'</span>
<span class="co">#&gt; This message will be shown once per session</span>
<span class="co">#&gt; Warning: Invalid name supplied, making object name syntactically valid. New</span>
<span class="co">#&gt; object name is RunUMAP.RNA.dr.sc; see ?make.names for more details on syntax</span>
<span class="co">#&gt; validity</span></code></pre></div>
<p><img src="DR.SC.DLPFC_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="fit-dr-sc-model-using-480-spatially-variable-features">Fit DR-SC model using 480 spatially variable features<a class="anchor" aria-label="anchor" href="#fit-dr-sc-model-using-480-spatially-variable-features"></a>
</h2>
<p>In spatially resolved transcriptomics data analysis, we recommend users using the spatially variable genes for analysis. We embeded the method SPARK-X (developed by Xiang Zhou’s Lab) into DR.SC package, which can be called using <code>FindSVGs</code>. The selected genes’ names are also saved in “<a href="mailto:seus@assays%24RNA@var.features" class="email">seus@assays$RNA@var.features</a>” and the order is determined by the statistical significance, where the gene with highest significance ranks first. We note there are some difference between SVGs and HVGs.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># choose 480 spatially variable features</span>
<span class="va">seus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindSVGs.html">FindSVGs</a></span><span class="op">(</span><span class="va">seu</span>, nfeatures <span class="op">=</span> <span class="fl">480</span><span class="op">)</span>
<span class="co">#&gt; Find the spatially variables genes by SPARK-X...</span>
<span class="co">#&gt; ## ===== SPARK-X INPUT INFORMATION ====</span>
<span class="co">#&gt; ## number of total samples: 4634</span>
<span class="co">#&gt; ## number of total genes: 500</span>
<span class="co">#&gt; ## Running with single core, may take some time</span>
<span class="co">#&gt; ## Testing With Projection Kernel</span>
<span class="co">#&gt; ## Testing With Gaussian Kernel 1</span>
<span class="co">#&gt; ## Testing With Gaussian Kernel 2</span>
<span class="co">#&gt; ## Testing With Gaussian Kernel 3</span>
<span class="co">#&gt; ## Testing With Gaussian Kernel 4</span>
<span class="co">#&gt; ## Testing With Gaussian Kernel 5</span>
<span class="co">#&gt; ## Testing With Cosine Kernel 1</span>
<span class="co">#&gt; ## Testing With Cosine Kernel 2</span>
<span class="co">#&gt; ## Testing With Cosine Kernel 3</span>
<span class="co">#&gt; ## Testing With Cosine Kernel 4</span>
<span class="co">#&gt; ## Testing With Cosine Kernel 5</span>
<span class="va">seus</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">var.features</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>
<span class="co">#&gt;  [1] "ENSG00000197971" "ENSG00000110484" "ENSG00000123560" "ENSG00000198804"</span>
<span class="co">#&gt;  [5] "ENSG00000124935" "ENSG00000173786" "ENSG00000168314" "ENSG00000198727"</span>
<span class="co">#&gt;  [9] "ENSG00000173432" "ENSG00000091513"</span></code></pre></div>
<p>We fit DR-SC model by using the selected spatially variable genes.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### Given K</span>
<span class="va">seus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DR.SC.html">DR.SC</a></span><span class="op">(</span><span class="va">seus</span>, K<span class="op">=</span><span class="fl">7</span>, platform <span class="op">=</span> <span class="st">'Visium'</span>, verbose<span class="op">=</span><span class="cn">T</span><span class="op">)</span>
<span class="co">#&gt; Neighbors were identified for 4634 out of 4634 spots.</span>
<span class="co">#&gt; Fit DR-SC model...</span>
<span class="co">#&gt; -------------------Calculate inital values-------------</span>
<span class="co">#&gt; -------------------Finish computing inital values-------------</span>
<span class="co">#&gt; -------------------Starting  ICM-EM algortihm-------------</span>
<span class="co">#&gt; iter = 2, loglik= -1228522.642564, dloglik=0.999428 </span>
<span class="co">#&gt; iter = 3, loglik= -1222070.073574, dloglik=0.005252 </span>
<span class="co">#&gt; iter = 4, loglik= -1220229.244307, dloglik=0.001506 </span>
<span class="co">#&gt; iter = 5, loglik= -1219175.850470, dloglik=0.000863 </span>
<span class="co">#&gt; iter = 6, loglik= -1218496.647196, dloglik=0.000557 </span>
<span class="co">#&gt; iter = 7, loglik= -1217959.809427, dloglik=0.000441 </span>
<span class="co">#&gt; iter = 8, loglik= -1217552.576343, dloglik=0.000334 </span>
<span class="co">#&gt; iter = 9, loglik= -1217209.187820, dloglik=0.000282 </span>
<span class="co">#&gt; iter = 10, loglik= -1216928.461434, dloglik=0.000231 </span>
<span class="co">#&gt; iter = 11, loglik= -1216689.528325, dloglik=0.000196 </span>
<span class="co">#&gt; iter = 12, loglik= -1216479.388650, dloglik=0.000173 </span>
<span class="co">#&gt; iter = 13, loglik= -1216298.431104, dloglik=0.000149 </span>
<span class="co">#&gt; iter = 14, loglik= -1216134.346168, dloglik=0.000135 </span>
<span class="co">#&gt; iter = 15, loglik= -1215985.318437, dloglik=0.000123 </span>
<span class="co">#&gt; iter = 16, loglik= -1215854.370103, dloglik=0.000108 </span>
<span class="co">#&gt; iter = 17, loglik= -1215739.339004, dloglik=0.000095 </span>
<span class="co">#&gt; iter = 18, loglik= -1215634.316515, dloglik=0.000086 </span>
<span class="co">#&gt; iter = 19, loglik= -1215526.012082, dloglik=0.000089 </span>
<span class="co">#&gt; iter = 20, loglik= -1215417.093982, dloglik=0.000090 </span>
<span class="co">#&gt; iter = 21, loglik= -1215331.186944, dloglik=0.000071 </span>
<span class="co">#&gt; iter = 22, loglik= -1215249.364654, dloglik=0.000067 </span>
<span class="co">#&gt; iter = 23, loglik= -1215167.058207, dloglik=0.000068 </span>
<span class="co">#&gt; iter = 24, loglik= -1215100.725954, dloglik=0.000055 </span>
<span class="co">#&gt; iter = 25, loglik= -1215037.260550, dloglik=0.000052</span>
<span class="co">#&gt; -------------------Complete!-------------</span>
<span class="co">#&gt; elasped time is :59.06</span>
<span class="co">#&gt; Finish DR-SC model fitting</span></code></pre></div>
<div class="section level3">
<h3 id="visualization-1">Visualization<a class="anchor" aria-label="anchor" href="#visualization-1"></a>
</h3>
<p>Next, we show the application of DR-SC in visualization. First, we can visualize the clusters from DR-SC on the spatial coordinates.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/spatialPlotClusters.html">spatialPlotClusters</a></span><span class="op">(</span><span class="va">seus</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.DLPFC_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>We can also visualize the clusters from DR-SC on the two-dimensional tSNE based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/drscPlot.html">drscPlot</a></span><span class="op">(</span><span class="va">seus</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.DLPFC_files/figure-html/unnamed-chunk-13-1.png" width="700"> Similarly, can also visualize the clusters from DR-SC on the two-dimensional UMAP based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/drscPlot.html">drscPlot</a></span><span class="op">(</span><span class="va">seus</span>, visu.method <span class="op">=</span> <span class="st">'UMAP'</span><span class="op">)</span>
<span class="co">#&gt; Warning: Invalid name supplied, making object name syntactically valid. New</span>
<span class="co">#&gt; object name is RunUMAP.RNA.dr.sc; see ?make.names for more details on syntax</span>
<span class="co">#&gt; validity</span></code></pre></div>
<p><img src="DR.SC.DLPFC_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p>Since DR.SC uses the Seurat object to save results, all visualization functions in Seurat package can used to visualize the results of DR-SC, such as ridge plot, feature plot, dot plot and so on. ### Ridge plots we show the application of DR-SC in differential expression analysis. Find the marker genes in SVGs for each clusters.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">SVGs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/topSVGs.html">topSVGs</a></span><span class="op">(</span><span class="va">seus</span>, ntop <span class="op">=</span> <span class="fl">400</span><span class="op">)</span>
<span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html" class="external-link">FindAllMarkers</a></span><span class="op">(</span><span class="va">seus</span>, features <span class="op">=</span> <span class="va">SVGs</span><span class="op">)</span>
<span class="co">#&gt; Calculating cluster cluster1</span>
<span class="co">#&gt; Calculating cluster cluster2</span>
<span class="co">#&gt; Calculating cluster cluster3</span>
<span class="co">#&gt; Calculating cluster cluster4</span>
<span class="co">#&gt; Calculating cluster cluster5</span>
<span class="co">#&gt; Calculating cluster cluster6</span>
<span class="co">#&gt; Calculating cluster cluster7</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span>
<span class="co">#&gt;                         p_val avg_log2FC pct.1 pct.2     p_val_adj  cluster</span>
<span class="co">#&gt; ENSG00000110484 6.391976e-207 -2.8930351 0.370 0.806 3.195988e-204 cluster1</span>
<span class="co">#&gt; ENSG00000124935 1.026265e-126 -2.4420247 0.186 0.613 5.131327e-124 cluster1</span>
<span class="co">#&gt; ENSG00000171617  1.987680e-91  0.7556269 0.963 0.772  9.938401e-89 cluster1</span>
<span class="co">#&gt; ENSG00000115756  4.320795e-76  0.9032286 0.748 0.423  2.160398e-73 cluster1</span>
<span class="co">#&gt; ENSG00000185499  8.148238e-66 -1.6087997 0.189 0.489  4.074119e-63 cluster1</span>
<span class="co">#&gt; ENSG00000162545  6.860146e-62  0.4809817 0.987 0.900  3.430073e-59 cluster1</span>
<span class="co">#&gt;                            gene</span>
<span class="co">#&gt; ENSG00000110484 ENSG00000110484</span>
<span class="co">#&gt; ENSG00000124935 ENSG00000124935</span>
<span class="co">#&gt; ENSG00000171617 ENSG00000171617</span>
<span class="co">#&gt; ENSG00000115756 ENSG00000115756</span>
<span class="co">#&gt; ENSG00000185499 ENSG00000185499</span>
<span class="co">#&gt; ENSG00000162545 ENSG00000162545</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.1.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="va">top2</span> <span class="op">&lt;-</span>  <span class="va">dat</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_n</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">2</span>, wt <span class="op">=</span> <span class="va">avg_log2FC</span><span class="op">)</span>
<span class="va">top2</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 14 x 7</span></span>
<span class="co">#&gt; <span style="color: #949494;"># Groups:   cluster [7]</span></span>
<span class="co">#&gt;        p_val avg_log2FC pct.1 pct.2 p_val_adj cluster  gene           </span>
<span class="co">#&gt;        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> 1.99<span style="color: #949494;">e</span><span style="color: #BB0000;">- 91</span>      0.756 0.963 0.772 9.94<span style="color: #949494;">e</span><span style="color: #BB0000;">- 89</span> cluster1 ENSG00000171617</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 4.32<span style="color: #949494;">e</span><span style="color: #BB0000;">- 76</span>      0.903 0.748 0.423 2.16<span style="color: #949494;">e</span><span style="color: #BB0000;">- 73</span> cluster1 ENSG00000115756</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> 1.49<span style="color: #949494;">e</span><span style="color: #BB0000;">-126</span>      3.69  0.995 0.79  7.47<span style="color: #949494;">e</span><span style="color: #BB0000;">-124</span> cluster2 ENSG00000197971</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> 1.29<span style="color: #949494;">e</span><span style="color: #BB0000;">-114</span>      3.70  0.963 0.639 6.44<span style="color: #949494;">e</span><span style="color: #BB0000;">-112</span> cluster2 ENSG00000123560</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> 8.04<span style="color: #949494;">e</span><span style="color: #BB0000;">-177</span>      0.962 0.975 0.613 4.02<span style="color: #949494;">e</span><span style="color: #BB0000;">-174</span> cluster3 ENSG00000110484</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> 1.58<span style="color: #949494;">e</span><span style="color: #BB0000;">-145</span>      0.977 0.848 0.393 7.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-143</span> cluster3 ENSG00000124935</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> 1.65<span style="color: #949494;">e</span><span style="color: #BB0000;">-254</span>      2.45  0.91  0.65  8.26<span style="color: #949494;">e</span><span style="color: #BB0000;">-252</span> cluster4 ENSG00000131095</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> 4.27<span style="color: #949494;">e</span><span style="color: #BB0000;">- 66</span>      1.88  0.472 0.261 2.14<span style="color: #949494;">e</span><span style="color: #BB0000;">- 63</span> cluster4 ENSG00000171885</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> 6.37<span style="color: #949494;">e</span><span style="color: #BB0000;">- 10</span>      0.621 0.359 0.25  3.19<span style="color: #949494;">e</span><span style="color: #BB0000;">-  7</span> cluster5 ENSG00000152377</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 5.79<span style="color: #949494;">e</span><span style="color: #BB0000;">-  8</span>      0.631 0.169 0.093 2.90<span style="color: #949494;">e</span><span style="color: #BB0000;">-  5</span> cluster5 ENSG00000158258</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">11</span> 4.88<span style="color: #949494;">e</span><span style="color: #BB0000;">- 43</span>      1.32  0.466 0.179 2.44<span style="color: #949494;">e</span><span style="color: #BB0000;">- 40</span> cluster6 ENSG00000183036</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">12</span> 4.89<span style="color: #949494;">e</span><span style="color: #BB0000;">- 40</span>      1.03  0.537 0.23  2.44<span style="color: #949494;">e</span><span style="color: #BB0000;">- 37</span> cluster6 ENSG00000155886</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">13</span> 9.58<span style="color: #949494;">e</span><span style="color: #BB0000;">- 36</span>      1.15  0.609 0.42  4.79<span style="color: #949494;">e</span><span style="color: #BB0000;">- 33</span> cluster7 ENSG00000145824</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">14</span> 1.47<span style="color: #949494;">e</span><span style="color: #BB0000;">-  6</span>      0.772 0.257 0.181 7.34<span style="color: #949494;">e</span><span style="color: #BB0000;">-  4</span> cluster7 ENSG00000173432</span></code></pre></div>
<p>Visualize single cell expression distributions in each cluster from Seruat.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">top2</span><span class="op">$</span><span class="va">gene</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">12</span>, by<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://satijalab.org/seurat/reference/RidgePlot.html" class="external-link">RidgePlot</a></span><span class="op">(</span><span class="va">seus</span>, features <span class="op">=</span> <span class="va">genes</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="co">#&gt; Picking joint bandwidth of 0.263</span>
<span class="co">#&gt; Picking joint bandwidth of 0.236</span>
<span class="co">#&gt; Picking joint bandwidth of 0.378</span>
<span class="co">#&gt; Picking joint bandwidth of 0.31</span>
<span class="co">#&gt; Picking joint bandwidth of 0.188</span>
<span class="co">#&gt; Picking joint bandwidth of 0.0977</span></code></pre></div>
<p><img src="DR.SC.DLPFC_files/figure-html/unnamed-chunk-16-1.png" width="960"> ### Violin plot</p>
<p>Visualize single cell expression distributions in each cluster</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://satijalab.org/seurat/reference/VlnPlot.html" class="external-link">VlnPlot</a></span><span class="op">(</span><span class="va">seus</span>, features <span class="op">=</span> <span class="va">genes</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.DLPFC_files/figure-html/unnamed-chunk-17-1.png" width="960"></p>
</div>
<div class="section level3">
<h3 id="feature-plot">Feature plot<a class="anchor" aria-label="anchor" href="#feature-plot"></a>
</h3>
<p>We extract tSNE based on the features from DR-SC and then visualize feature expression in the low-dimensional space</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seus</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunTSNE.html" class="external-link">RunTSNE</a></span><span class="op">(</span><span class="va">seus</span>, reduction<span class="op">=</span><span class="st">"dr-sc"</span>, reduction.key<span class="op">=</span><span class="st">'drsc_tSNE_'</span><span class="op">)</span>
<span class="co">#&gt; Warning: Keys should be one or more alphanumeric characters followed by an</span>
<span class="co">#&gt; underscore, setting key from drsc_tSNE_ to drsctSNE_</span>
<span class="co">#&gt; Warning: All keys should be one or more alphanumeric characters followed by an</span>
<span class="co">#&gt; underscore '_', setting key to drsctSNE_</span>
<span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">seus</span>, features <span class="op">=</span> <span class="va">genes</span>, reduction <span class="op">=</span> <span class="st">'tsne'</span> ,ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.DLPFC_files/figure-html/unnamed-chunk-18-1.png" width="960"></p>
</div>
<div class="section level3">
<h3 id="dot-plots">Dot plots<a class="anchor" aria-label="anchor" href="#dot-plots"></a>
</h3>
<p>The size of the dot corresponds to the percentage of cells expressing the feature in each cluster. The color represents the average expression level</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://satijalab.org/seurat/reference/DotPlot.html" class="external-link">DotPlot</a></span><span class="op">(</span><span class="va">seus</span>, features <span class="op">=</span> <span class="va">genes</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.DLPFC_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="heatmap-plot">Heatmap plot<a class="anchor" aria-label="anchor" href="#heatmap-plot"></a>
</h3>
<p>Single cell heatmap of feature expression</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">top20</span> <span class="op">&lt;-</span>  <span class="va">dat</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_n</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">20</span>, wt <span class="op">=</span> <span class="va">avg_log2FC</span><span class="op">)</span>
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">top20</span><span class="op">$</span><span class="va">gene</span>
<span class="co"># standard scaling (no regression)</span>
<span class="va">seus</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">seus</span><span class="op">)</span>
<span class="co">#&gt; Centering and scaling data matrix</span>
<span class="fu"><a href="https://satijalab.org/seurat/reference/DoHeatmap.html" class="external-link">DoHeatmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">seus</span>, downsample <span class="op">=</span> <span class="fl">500</span><span class="op">)</span>, features <span class="op">=</span> <span class="va">genes</span>, size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>
<span class="co">#&gt; Warning: Invalid name supplied, making object name syntactically valid. New</span>
<span class="co">#&gt; object name is dr.sc; see ?make.names for more details on syntax validity</span>
<span class="co">#&gt; Warning: Cannot add objects with duplicate keys (offending key: DRSC_), setting</span>
<span class="co">#&gt; key to 'dr.sc_'</span></code></pre></div>
<p><img src="DR.SC.DLPFC_files/figure-html/unnamed-chunk-20-1.png" width="960"></p>
</div>
</div>
<div class="section level2">
<h2 id="fit-dr-sc-model-using-480-spatially-variable-features-and-using-mbic-to-determine-clusters">Fit DR-SC model using 480 spatially variable features and using MBIC　to determine clusters<a class="anchor" aria-label="anchor" href="#fit-dr-sc-model-using-480-spatially-variable-features-and-using-mbic-to-determine-clusters"></a>
</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># choose spatially variable features</span>
<span class="va">seus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindSVGs.html">FindSVGs</a></span><span class="op">(</span><span class="va">seu</span>, nfeatures <span class="op">=</span> <span class="fl">480</span>, verbose <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<p>We set the argument variable.type=‘SVGs’ (default option) to use the spatially variable genes.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### Given K</span>
<span class="va">seus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DR.SC.html">DR.SC</a></span><span class="op">(</span><span class="va">seus</span>, K<span class="op">=</span><span class="fl">3</span><span class="op">:</span><span class="fl">9</span>, platform <span class="op">=</span> <span class="st">'Visium'</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="co">#&gt; Neighbors were identified for 4634 out of 4634 spots.</span>
<span class="co">#&gt; Fit DR-SC model...</span>
<span class="co">#&gt; Starting parallel computing intial values...</span>
<span class="co">#&gt; Finish DR-SC model fitting</span></code></pre></div>
<p>Plot the MBIC curve</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selectModel.html">selectModel</a></span><span class="op">(</span><span class="va">seus</span>, pen.const <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span>
<span class="fu"><a href="../reference/mbicPlot.html">mbicPlot</a></span><span class="op">(</span><span class="va">seus</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.DLPFC_files/figure-html/unnamed-chunk-23-1.png" width="700"></p>
<p>Show the spatial scatter plot for clusters</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/spatialPlotClusters.html">spatialPlotClusters</a></span><span class="op">(</span><span class="va">seus</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.DLPFC_files/figure-html/unnamed-chunk-24-1.png" width="700"></p>
<p>Show the tSNE plot based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/drscPlot.html">drscPlot</a></span><span class="op">(</span><span class="va">seus</span>, dims<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.DLPFC_files/figure-html/unnamed-chunk-25-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span>
<span class="co">#&gt; Running under: Windows 10 x64 (build 22000)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] LC_COLLATE=English_Singapore.1252  LC_CTYPE=English_Singapore.1252   </span>
<span class="co">#&gt; [3] LC_MONETARY=English_Singapore.1252 LC_NUMERIC=C                      </span>
<span class="co">#&gt; [5] LC_TIME=English_Singapore.1252    </span>
<span class="co">#&gt; system code page: 936</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] dplyr_1.0.9         sp_1.5-0            SeuratObject_4.1.0 </span>
<span class="co">#&gt; [4] Seurat_4.1.1        DR.SC_2.9           spatstat.geom_2.4-0</span>
<span class="co">#&gt; [7] spatstat.data_2.2-0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] GiRaF_1.0.1           systemfonts_1.0.4     plyr_1.8.7           </span>
<span class="co">#&gt;   [4] igraph_1.3.2          lazyeval_0.2.2        splines_4.1.2        </span>
<span class="co">#&gt;   [7] listenv_0.8.0         scattermore_0.8       ggplot2_3.3.6        </span>
<span class="co">#&gt;  [10] digest_0.6.29         htmltools_0.5.2       fansi_1.0.3          </span>
<span class="co">#&gt;  [13] magrittr_2.0.3        memoise_2.0.1         tensor_1.5           </span>
<span class="co">#&gt;  [16] cluster_2.1.2         ROCR_1.0-11           limma_3.50.1         </span>
<span class="co">#&gt;  [19] globals_0.15.0        matrixStats_0.62.0    pkgdown_2.0.2        </span>
<span class="co">#&gt;  [22] spatstat.sparse_2.1-1 colorspace_2.0-3      ggrepel_0.9.1        </span>
<span class="co">#&gt;  [25] textshaping_0.3.6     xfun_0.29             crayon_1.5.1         </span>
<span class="co">#&gt;  [28] jsonlite_1.8.0        progressr_0.10.1      survival_3.2-13      </span>
<span class="co">#&gt;  [31] zoo_1.8-10            glue_1.6.2            polyclip_1.10-0      </span>
<span class="co">#&gt;  [34] gtable_0.3.0          leiden_0.4.2          future.apply_1.9.0   </span>
<span class="co">#&gt;  [37] BiocGenerics_0.40.0   abind_1.4-5           scales_1.2.0         </span>
<span class="co">#&gt;  [40] DBI_1.1.2             spatstat.random_2.2-0 miniUI_0.1.1.1       </span>
<span class="co">#&gt;  [43] Rcpp_1.0.8.3          viridisLite_0.4.0     xtable_1.8-4         </span>
<span class="co">#&gt;  [46] reticulate_1.25       spatstat.core_2.4-4   mclust_5.4.10        </span>
<span class="co">#&gt;  [49] stats4_4.1.2          htmlwidgets_1.5.4     httr_1.4.3           </span>
<span class="co">#&gt;  [52] RColorBrewer_1.1-3    ellipsis_0.3.2        ica_1.0-2            </span>
<span class="co">#&gt;  [55] pkgconfig_2.0.3       farver_2.1.0          sass_0.4.1           </span>
<span class="co">#&gt;  [58] uwot_0.1.11           deldir_1.0-6          utf8_1.2.2           </span>
<span class="co">#&gt;  [61] tidyselect_1.1.2      labeling_0.4.2        rlang_1.0.2          </span>
<span class="co">#&gt;  [64] reshape2_1.4.4        later_1.3.0           munsell_0.5.0        </span>
<span class="co">#&gt;  [67] tools_4.1.2           cachem_1.0.6          cli_3.2.0            </span>
<span class="co">#&gt;  [70] generics_0.1.2        ggridges_0.5.3        evaluate_0.15        </span>
<span class="co">#&gt;  [73] stringr_1.4.0         fastmap_1.1.0         yaml_2.3.5           </span>
<span class="co">#&gt;  [76] ragg_1.2.2            goftest_1.2-3         knitr_1.37           </span>
<span class="co">#&gt;  [79] fs_1.5.2              fitdistrplus_1.1-8    purrr_0.3.4          </span>
<span class="co">#&gt;  [82] RANN_2.6.1            pbapply_1.5-0         future_1.26.1        </span>
<span class="co">#&gt;  [85] nlme_3.1-155          mime_0.12             ggrastr_1.0.1        </span>
<span class="co">#&gt;  [88] compiler_4.1.2        rstudioapi_0.13       beeswarm_0.4.0       </span>
<span class="co">#&gt;  [91] plotly_4.10.0         png_0.1-7             spatstat.utils_2.3-1 </span>
<span class="co">#&gt;  [94] tibble_3.1.7          bslib_0.3.1           stringi_1.7.6        </span>
<span class="co">#&gt;  [97] highr_0.9             RSpectra_0.16-1       desc_1.4.0           </span>
<span class="co">#&gt; [100] rgeos_0.5-9           lattice_0.20-45       Matrix_1.4-0         </span>
<span class="co">#&gt; [103] vctrs_0.4.1           CompQuadForm_1.4.3    pillar_1.7.0         </span>
<span class="co">#&gt; [106] lifecycle_1.0.1       lmtest_0.9-40         jquerylib_0.1.4      </span>
<span class="co">#&gt; [109] RcppAnnoy_0.0.19      data.table_1.14.2     cowplot_1.1.1        </span>
<span class="co">#&gt; [112] irlba_2.3.5           httpuv_1.6.5          patchwork_1.1.1      </span>
<span class="co">#&gt; [115] R6_2.5.1              promises_1.2.0.1      KernSmooth_2.23-20   </span>
<span class="co">#&gt; [118] gridExtra_2.3         vipor_0.4.5           parallelly_1.32.0    </span>
<span class="co">#&gt; [121] codetools_0.2-18      MASS_7.3-55           assertthat_0.2.1     </span>
<span class="co">#&gt; [124] rprojroot_2.0.3       withr_2.5.0           sctransform_0.3.3    </span>
<span class="co">#&gt; [127] S4Vectors_0.32.3      mgcv_1.8-39           grid_4.1.2           </span>
<span class="co">#&gt; [130] rpart_4.1.16          tidyr_1.2.0           rmarkdown_2.11       </span>
<span class="co">#&gt; [133] Rtsne_0.16            shiny_1.7.1           ggbeeswarm_0.6.0</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Wei Liu, Yi Yang, Jin Liu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
