<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>DR-SC: installation and simulation â€¢ DR.SC</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="DR-SC: installation and simulation">
<meta property="og:description" content="DR.SC">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DR.SC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/DR.SC.HCC1.html">DR-SC: HCC1 Data Analysis</a>
    </li>
    <li>
      <a href="../articles/DR.SC.Simu.html">DR-SC: installation and simulation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>DR-SC: installation and simulation</h1>
                        <h4 class="author">Wei Liu</h4>
            
            <h4 class="date">2021-10-06</h4>
      
      
      <div class="hidden name"><code>DR.SC.Simu.Rmd</code></div>

    </div>

    
    
<div id="install-the-dr-sc" class="section level2">
<h2 class="hasAnchor">
<a href="#install-the-dr-sc" class="anchor"></a>Install the DR.SC</h2>
<p>This vignette provides an introduction to the R package <code>DR.SC</code>, where the function <code>DR.SC</code> implements the model <code>DR-SC</code>, spatial clustering with hidden Markov random field using empirical Bayes. The package can be installed with the command:</p>
<p><code><a href="https://remotes.r-lib.org">library(remotes)</a></code></p>
<p><code><a href="https://remotes.r-lib.org/reference/install_github.html">remotes::install_github("feiyoung/DR.SC")</a></code></p>
<p>The package can be loaded with the command:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"DR.SC"</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: mclust</span>
<span class="co">#&gt; Package 'mclust' version 5.4.7</span>
<span class="co">#&gt; Type 'citation("mclust")' for citing this R package in publications.</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Loading required package: SingleCellExperiment</span>
<span class="co">#&gt; Loading required package: SummarizedExperiment</span>
<span class="co">#&gt; Loading required package: MatrixGenerics</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'MatrixGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span>
<span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span>
<span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span>
<span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span>
<span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span>
<span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span>
<span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span>
<span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span>
<span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span>
<span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span>
<span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span>
<span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span>
<span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span>
<span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span>
<span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span>
<span class="co">#&gt; Loading required package: GenomicRanges</span>
<span class="co">#&gt; Loading required package: stats4</span>
<span class="co">#&gt; Loading required package: BiocGenerics</span>
<span class="co">#&gt; Warning: package 'BiocGenerics' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'BiocGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:parallel':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
<span class="co">#&gt;     clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
<span class="co">#&gt;     parLapplyLB, parRapply, parSapply, parSapplyLB</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
<span class="co">#&gt;     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
<span class="co">#&gt;     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
<span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
<span class="co">#&gt;     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
<span class="co">#&gt;     union, unique, unsplit, which.max, which.min</span>
<span class="co">#&gt; Loading required package: S4Vectors</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'S4Vectors'</span>
<span class="co">#&gt; The following object is masked from 'package:Matrix':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand.grid</span>
<span class="co">#&gt; Loading required package: IRanges</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'IRanges'</span>
<span class="co">#&gt; The following object is masked from 'package:grDevices':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     windows</span>
<span class="co">#&gt; Loading required package: GenomeInfoDb</span>
<span class="co">#&gt; Warning: package 'GenomeInfoDb' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: Biobase</span>
<span class="co">#&gt; Welcome to Bioconductor</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Vignettes contain introductory material; view with</span>
<span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'Biobase'</span>
<span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     rowMedians</span>
<span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyMissing, rowMedians</span>
<span class="co">#&gt; Loading required package: MASS</span>
<span class="co">#&gt; Warning: package 'MASS' was built under R version 4.0.4</span>
<span class="co">#&gt; Loading required package: purrr</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'purrr'</span>
<span class="co">#&gt; The following object is masked from 'package:GenomicRanges':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     reduce</span>
<span class="co">#&gt; The following object is masked from 'package:IRanges':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     reduce</span>
<span class="co">#&gt; The following object is masked from 'package:mclust':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     map</span>
<span class="co">#&gt; Loading required package: doSNOW</span>
<span class="co">#&gt; Warning: package 'doSNOW' was built under R version 4.0.5</span>
<span class="co">#&gt; Loading required package: foreach</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'foreach'</span>
<span class="co">#&gt; The following objects are masked from 'package:purrr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     accumulate, when</span>
<span class="co">#&gt; Loading required package: iterators</span>
<span class="co">#&gt; Loading required package: snow</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'snow'</span>
<span class="co">#&gt; The following objects are masked from 'package:BiocGenerics':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
<span class="co">#&gt;     clusterExport, clusterMap, clusterSplit, parApply, parCapply,</span>
<span class="co">#&gt;     parLapply, parRapply, parSapply</span>
<span class="co">#&gt; The following objects are masked from 'package:parallel':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
<span class="co">#&gt;     clusterExport, clusterMap, clusterSplit, makeCluster, parApply,</span>
<span class="co">#&gt;     parCapply, parLapply, parRapply, parSapply, splitIndices,</span>
<span class="co">#&gt;     stopCluster</span>
<span class="co">#&gt; Loading required package: Seurat</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'Seurat'</span>
<span class="co">#&gt; The following object is masked from 'package:SummarizedExperiment':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Assays</span>
<span class="co">#&gt; Loading required package: RColorBrewer</span>
<span class="co">#&gt; Loading required package: ggplot2</span></code></pre></div>
</div>
<div id="fit-dr-sc-using-simulated-data" class="section level2">
<h2 class="hasAnchor">
<a href="#fit-dr-sc-using-simulated-data" class="anchor"></a>Fit DR-SC using simulated data</h2>
<div id="generating-the-simulated-data" class="section level3">
<h3 class="hasAnchor">
<a href="#generating-the-simulated-data" class="anchor"></a>Generating the simulated data</h3>
<p>First, we generate the spatial transcriptomics data with lattice neighborhood, i.e.Â ST platform.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gendata_RNAExp.html">gendata_RNAExp</a></span><span class="op">(</span>height<span class="op">=</span><span class="fl">30</span>, width<span class="op">=</span><span class="fl">30</span>,p<span class="op">=</span><span class="fl">500</span>, K<span class="op">=</span><span class="fl">4</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: GiRaF</span>
<span class="co">#&gt; Warning: package 'GiRaF' was built under R version 4.0.4</span>
<span class="co">#&gt; Warning: The following arguments are not used: row.names</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">seu</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span>
<span class="co">#&gt;           orig.ident nCount_RNA nFeature_RNA row col imagerow imagecol</span>
<span class="co">#&gt; spot-1 SeuratProject    1631885          500   1   1        1        1</span>
<span class="co">#&gt; spot-2 SeuratProject    2138381          500   2   1        2        1</span>
<span class="co">#&gt; spot-3 SeuratProject    1867949          500   3   1        3        1</span>
<span class="co">#&gt; spot-4 SeuratProject    3021387          500   4   1        4        1</span>
<span class="co">#&gt; spot-5 SeuratProject    2590291          500   5   1        5        1</span>
<span class="co">#&gt; spot-6 SeuratProject    2995910          500   6   1        6        1</span>
<span class="co">#&gt;        true_clusters</span>
<span class="co">#&gt; spot-1             1</span>
<span class="co">#&gt; spot-2             1</span>
<span class="co">#&gt; spot-3             1</span>
<span class="co">#&gt; spot-4             4</span>
<span class="co">#&gt; spot-5             4</span>
<span class="co">#&gt; spot-6             4</span></code></pre></div>
</div>
<div id="fitting-model-by-using-dr-sc-function" class="section level3">
<h3 class="hasAnchor">
<a href="#fitting-model-by-using-dr-sc-function" class="anchor"></a>Fitting model by using DR.SC function</h3>
<p>For function DR.SC, users can specify the number of clusters <span class="math inline">\(K\)</span> or set {K=NULL} by using modified BIC(MBIC) to determine <span class="math inline">\(K\)</span>. First, we try using user-specified number of clusters. Then we show the version chosen by MBIC.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### Given K</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html">NormalizeData</a></span><span class="op">(</span><span class="va">seu</span><span class="op">)</span>
<span class="co"># choose 2000 variable features using Seurat</span>
<span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html">FindVariableFeatures</a></span><span class="op">(</span><span class="va">seu</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span>
<span class="va">seu2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DR.SC.html">DR.SC</a></span><span class="op">(</span><span class="va">seu</span>, K<span class="op">=</span><span class="fl">4</span>, platform <span class="op">=</span> <span class="st">'ST'</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="co">#&gt; Neighbors were identified for 900 out of 900 spots.</span>
<span class="co">#&gt; Warning in DR.SC.Seurat(seu, K = 4, platform = "ST", verbose = F): nrow(seu) is</span>
<span class="co">#&gt; less than nfeatures, so assign nfeatures with nrow(seu)!</span>
<span class="co">#&gt; Fit DR-SC model...</span>
<span class="co">#&gt; elasped time is : 9.76 0.73 10.33 NA NA </span>
<span class="co">#&gt; Finish DR-SC model fitting</span></code></pre></div>
<p>Using ARI to check the performance of clustering</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">mclust</span><span class="fu">::</span><span class="fu"><a href="https://mclust-org.github.io/mclust/reference/adjustedRandIndex.html">adjustedRandIndex</a></span><span class="op">(</span><span class="va">seu2</span><span class="op">$</span><span class="va">spatial.drsc.cluster</span>, <span class="va">seu</span><span class="op">$</span><span class="va">true_clusters</span><span class="op">)</span>
<span class="co">#&gt; [1] 1</span></code></pre></div>
<p>Show the spatial scatter plot for clusters</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/spatialPlotClusters.html">spatialPlotClusters</a></span><span class="op">(</span><span class="va">seu2</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<p>Show the tSNE plot based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/drscPlot.html">drscPlot</a></span><span class="op">(</span><span class="va">seu2</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-6-1.png" width="700"> Show the UMAP plot based on the extracted features from DR-SC.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/drscPlot.html">drscPlot</a></span><span class="op">(</span><span class="va">seu2</span>, visu.method <span class="op">=</span> <span class="st">'UMAP'</span><span class="op">)</span>
<span class="co">#&gt; Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric</span>
<span class="co">#&gt; To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'</span>
<span class="co">#&gt; This message will be shown once per session</span>
<span class="co">#&gt; Warning: Invalid name supplied, making object name syntactically valid. New</span>
<span class="co">#&gt; object name is RunUMAP.RNA.dr.sc; see ?make.names for more details on syntax</span>
<span class="co">#&gt; validity</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>Use MBIC to choose number of clusters:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" title="1">seu2 &lt;-<span class="st"> </span><span class="kw">DR.SC</span>(seu, <span class="dt">q=</span><span class="dv">10</span>, <span class="dt">K=</span><span class="ot">NULL</span>, <span class="dt">K_set =</span><span class="dv">2</span><span class="op">:</span><span class="dv">6</span>, <span class="dt">platform =</span> <span class="st">'ST'</span>, <span class="dt">verbose=</span>F)</a>
<a class="sourceLine" id="cb8-2" title="2"><span class="co">#&gt; Neighbors were identified for 900 out of 900 spots.</span></a>
<a class="sourceLine" id="cb8-3" title="3"><span class="co">#&gt; Warning in DR.SC.Seurat(seu, q = 10, K = NULL, K_set = 2:6, platform = "ST", :</span></a>
<a class="sourceLine" id="cb8-4" title="4"><span class="co">#&gt; nrow(seu) is less than nfeatures, so assign nfeatures with nrow(seu)!</span></a>
<a class="sourceLine" id="cb8-5" title="5"><span class="co">#&gt; Start chooing number of clusters...</span></a>
<a class="sourceLine" id="cb8-6" title="6"><span class="co">#&gt; The candidate set is:  2 3 4 5 6 </span></a>
<a class="sourceLine" id="cb8-7" title="7"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb8-8" title="8">  <span class="op">|</span><span class="st">                                                                            </span></a>
<a class="sourceLine" id="cb8-9" title="9"><span class="st">  </span><span class="er">|</span><span class="st">                                                                      </span><span class="er">|</span><span class="st">   </span><span class="dv">0</span>%</a>
<a class="sourceLine" id="cb8-10" title="10">  <span class="op">|</span><span class="st">                                                                            </span></a>
<a class="sourceLine" id="cb8-11" title="11"><span class="st">  </span><span class="er">|==================</span><span class="st">                                                    </span><span class="er">|</span><span class="st">  </span><span class="dv">25</span>%</a>
<a class="sourceLine" id="cb8-12" title="12">  <span class="op">|</span><span class="st">                                                                            </span></a>
<a class="sourceLine" id="cb8-13" title="13"><span class="st">  </span><span class="er">|===================================</span><span class="st">                                   </span><span class="er">|</span><span class="st">  </span><span class="dv">50</span>%</a>
<a class="sourceLine" id="cb8-14" title="14">  <span class="op">|</span><span class="st">                                                                            </span></a>
<a class="sourceLine" id="cb8-15" title="15"><span class="st">  </span><span class="er">|====================================================</span><span class="st">                  </span><span class="er">|</span><span class="st">  </span><span class="dv">75</span>%</a>
<a class="sourceLine" id="cb8-16" title="16">  <span class="op">|</span><span class="st">                                                                            </span></a>
<a class="sourceLine" id="cb8-17" title="17"><span class="st">  </span><span class="er">|======================================================================|</span><span class="st"> </span><span class="dv">100</span>%</a>
<a class="sourceLine" id="cb8-18" title="18"><span class="co">#&gt; The best number of cluster is:  4 </span></a>
<a class="sourceLine" id="cb8-19" title="19"><span class="co">#&gt; Fit DR-SC model...</span></a>
<a class="sourceLine" id="cb8-20" title="20"><span class="co">#&gt; elasped time is : 6.18 1.33 8.19 NA NA </span></a>
<a class="sourceLine" id="cb8-21" title="21"><span class="co">#&gt; Finish DR-SC model fitting</span></a>
<a class="sourceLine" id="cb8-22" title="22"><span class="kw">mbicPlot</span>(seu2)</a></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
</div>
<div id="visualizations-of-marker-gene-expression" class="section level2">
<h2 class="hasAnchor">
<a href="#visualizations-of-marker-gene-expression" class="anchor"></a>Visualizations of marker gene expression</h2>
<div id="ridge-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#ridge-plots" class="anchor"></a>Ridge plots</h3>
<p>Visualize single cell expression distributions in each cluster from Seruat.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"gene-24"</span>,<span class="st">"gene-68"</span>, <span class="st">"gene-95"</span>,<span class="st">"gene-55"</span><span class="op">)</span>
<span class="fu"><a href="https://satijalab.org/seurat/reference/RidgePlot.html">RidgePlot</a></span><span class="op">(</span><span class="va">seu2</span>, features <span class="op">=</span> <span class="va">genes</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="co">#&gt; Picking joint bandwidth of 0.323</span>
<span class="co">#&gt; Picking joint bandwidth of 0.335</span>
<span class="co">#&gt; Picking joint bandwidth of 0.336</span>
<span class="co">#&gt; Picking joint bandwidth of 0.323</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-9-1.png" width="700"> ### Violin plot Visualize single cell expression distributions in each cluster</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://satijalab.org/seurat/reference/VlnPlot.html">VlnPlot</a></span><span class="op">(</span><span class="va">seu2</span>, features <span class="op">=</span> <span class="va">genes</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-10-1.png" width="700"> ### Feature plot We extract tSNE based on the features from DR-SC and then visualize feature expression in the low-dimensional space</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seu2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunTSNE.html">RunTSNE</a></span><span class="op">(</span><span class="va">seu2</span>, reduction<span class="op">=</span><span class="st">"dr-sc"</span>, reduction.key<span class="op">=</span><span class="st">'drsc_tSNE_'</span><span class="op">)</span>
<span class="co">#&gt; Warning: Keys should be one or more alphanumeric characters followed by an</span>
<span class="co">#&gt; underscore, setting key from drsc_tSNE_ to drsctSNE_</span>
<span class="co">#&gt; Warning: All keys should be one or more alphanumeric characters followed by an</span>
<span class="co">#&gt; underscore '_', setting key to drsctSNE_</span>
<span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html">FeaturePlot</a></span><span class="op">(</span><span class="va">seu2</span>, features <span class="op">=</span> <span class="va">genes</span>, reduction <span class="op">=</span> <span class="st">'tsne'</span> ,ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
</div>
<div id="dot-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#dot-plots" class="anchor"></a>Dot plots</h3>
<p>The size of the dot corresponds to the percentage of cells expressing the feature in each cluster. The color represents the average expression level</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://satijalab.org/seurat/reference/DotPlot.html">DotPlot</a></span><span class="op">(</span><span class="va">seu2</span>, features <span class="op">=</span> <span class="va">genes</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
</div>
<div id="heatmap-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#heatmap-plot" class="anchor"></a>Heatmap plot</h3>
<p>Single cell heatmap of feature expression</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># standard scaling (no regression)</span>
<span class="va">seu2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html">ScaleData</a></span><span class="op">(</span><span class="va">seu2</span><span class="op">)</span>
<span class="co">#&gt; Centering and scaling data matrix</span>
<span class="fu"><a href="https://satijalab.org/seurat/reference/DoHeatmap.html">DoHeatmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">seu2</span>, downsample <span class="op">=</span> <span class="fl">500</span><span class="op">)</span>, features <span class="op">=</span> <span class="va">genes</span>, size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></code></pre></div>
<p><img src="DR.SC.Simu_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
</div>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.3 (2020-10-10)</span>
<span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span>
<span class="co">#&gt; Running under: Windows 10 x64 (build 19043)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] LC_COLLATE=Chinese (Simplified)_China.936 </span>
<span class="co">#&gt; [2] LC_CTYPE=Chinese (Simplified)_China.936   </span>
<span class="co">#&gt; [3] LC_MONETARY=Chinese (Simplified)_China.936</span>
<span class="co">#&gt; [4] LC_NUMERIC=C                              </span>
<span class="co">#&gt; [5] LC_TIME=Chinese (Simplified)_China.936    </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] GiRaF_1.0.1                 DR.SC_2.2                  </span>
<span class="co">#&gt;  [3] ggplot2_3.3.5               RColorBrewer_1.1-2         </span>
<span class="co">#&gt;  [5] SeuratObject_4.0.2          Seurat_4.0.3               </span>
<span class="co">#&gt;  [7] doSNOW_1.0.19               snow_0.4-3                 </span>
<span class="co">#&gt;  [9] iterators_1.0.13            foreach_1.5.1              </span>
<span class="co">#&gt; [11] purrr_0.3.4                 MASS_7.3-53.1              </span>
<span class="co">#&gt; [13] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0</span>
<span class="co">#&gt; [15] Biobase_2.50.0              GenomicRanges_1.42.0       </span>
<span class="co">#&gt; [17] GenomeInfoDb_1.26.7         IRanges_2.24.1             </span>
<span class="co">#&gt; [19] S4Vectors_0.28.1            BiocGenerics_0.36.1        </span>
<span class="co">#&gt; [21] MatrixGenerics_1.2.1        matrixStats_0.58.0         </span>
<span class="co">#&gt; [23] Matrix_1.2-18               mclust_5.4.7               </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] systemfonts_1.0.2      plyr_1.8.6             igraph_1.2.6          </span>
<span class="co">#&gt;   [4] lazyeval_0.2.2         splines_4.0.3          listenv_0.8.0         </span>
<span class="co">#&gt;   [7] scattermore_0.7        digest_0.6.27          htmltools_0.5.1.1     </span>
<span class="co">#&gt;  [10] fansi_0.5.0            magrittr_2.0.1         memoise_2.0.0         </span>
<span class="co">#&gt;  [13] tensor_1.5             cluster_2.1.1          ROCR_1.0-11           </span>
<span class="co">#&gt;  [16] globals_0.14.0         pkgdown_1.6.1          spatstat.sparse_2.0-0 </span>
<span class="co">#&gt;  [19] colorspace_2.0-0       ggrepel_0.9.1          textshaping_0.3.5     </span>
<span class="co">#&gt;  [22] xfun_0.22              dplyr_1.0.5            crayon_1.4.1          </span>
<span class="co">#&gt;  [25] RCurl_1.98-1.2         jsonlite_1.7.2         spatstat.data_2.1-0   </span>
<span class="co">#&gt;  [28] survival_3.2-7         zoo_1.8-9              glue_1.4.2            </span>
<span class="co">#&gt;  [31] polyclip_1.10-0        gtable_0.3.0           zlibbioc_1.36.0       </span>
<span class="co">#&gt;  [34] XVector_0.30.0         leiden_0.3.7           DelayedArray_0.16.0   </span>
<span class="co">#&gt;  [37] future.apply_1.7.0     abind_1.4-5            scales_1.1.1          </span>
<span class="co">#&gt;  [40] DBI_1.1.1              miniUI_0.1.1.1         Rcpp_1.0.7            </span>
<span class="co">#&gt;  [43] viridisLite_0.4.0      xtable_1.8-4           reticulate_1.18       </span>
<span class="co">#&gt;  [46] spatstat.core_2.0-0    htmlwidgets_1.5.3      httr_1.4.2            </span>
<span class="co">#&gt;  [49] ellipsis_0.3.2         ica_1.0-2              farver_2.1.0          </span>
<span class="co">#&gt;  [52] pkgconfig_2.0.3        sass_0.3.1             uwot_0.1.10           </span>
<span class="co">#&gt;  [55] deldir_0.2-10          utf8_1.2.1             labeling_0.4.2        </span>
<span class="co">#&gt;  [58] tidyselect_1.1.1       rlang_0.4.11           reshape2_1.4.4        </span>
<span class="co">#&gt;  [61] later_1.1.0.1          munsell_0.5.0          tools_4.0.3           </span>
<span class="co">#&gt;  [64] cachem_1.0.4           generics_0.1.0         ggridges_0.5.3        </span>
<span class="co">#&gt;  [67] evaluate_0.14          stringr_1.4.0          fastmap_1.1.0         </span>
<span class="co">#&gt;  [70] yaml_2.2.1             ragg_1.1.3             goftest_1.2-2         </span>
<span class="co">#&gt;  [73] knitr_1.31             fs_1.5.0               fitdistrplus_1.1-3    </span>
<span class="co">#&gt;  [76] RANN_2.6.1             pbapply_1.4-3          future_1.21.0         </span>
<span class="co">#&gt;  [79] nlme_3.1-152           mime_0.10              compiler_4.0.3        </span>
<span class="co">#&gt;  [82] plotly_4.9.3           png_0.1-7              spatstat.utils_2.1-0  </span>
<span class="co">#&gt;  [85] tibble_3.1.2           bslib_0.2.4            stringi_1.7.3         </span>
<span class="co">#&gt;  [88] highr_0.8              desc_1.3.0             lattice_0.20-41       </span>
<span class="co">#&gt;  [91] vctrs_0.3.8            pillar_1.6.1           lifecycle_1.0.0       </span>
<span class="co">#&gt;  [94] spatstat.geom_2.0-1    lmtest_0.9-38          jquerylib_0.1.3       </span>
<span class="co">#&gt;  [97] RcppAnnoy_0.0.18       data.table_1.14.0      cowplot_1.1.1         </span>
<span class="co">#&gt; [100] bitops_1.0-6           irlba_2.3.3            httpuv_1.5.5          </span>
<span class="co">#&gt; [103] patchwork_1.1.1        R6_2.5.0               promises_1.2.0.1      </span>
<span class="co">#&gt; [106] KernSmooth_2.23-18     gridExtra_2.3          parallelly_1.23.0     </span>
<span class="co">#&gt; [109] codetools_0.2-18       assertthat_0.2.1       rprojroot_2.0.2       </span>
<span class="co">#&gt; [112] withr_2.4.2            sctransform_0.3.2      GenomeInfoDbData_1.2.4</span>
<span class="co">#&gt; [115] mgcv_1.8-34            grid_4.0.3             rpart_4.1-15          </span>
<span class="co">#&gt; [118] tidyr_1.1.3            rmarkdown_2.7          Rtsne_0.15            </span>
<span class="co">#&gt; [121] shiny_1.6.0</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Wei Liu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
