<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Joint dimension reduction and spatial clustering — DR.SC • DR.SC</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Joint dimension reduction and spatial clustering — DR.SC"><meta property="og:description" content="Joint dimension reduction and spatial clustering for scRNA-seq and spatial transcriptomics data"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DR.SC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/DR-SC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/DR.SC.DLPFC.html">DR-SC: DLPFC Data Analysis</a>
    </li>
    <li>
      <a href="../articles/DR.SC.Simu.html">DR-SC: simulation</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/feiyoung/DR.SC/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Joint dimension reduction and spatial clustering</h1>
    
    <div class="hidden name"><code>DR.SC.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Joint dimension reduction and spatial clustering for scRNA-seq and spatial transcriptomics data</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="co"># S3 method for Seurat</span>
<span class="fu">DR.SC</span><span class="op">(</span><span class="va">seu</span>, <span class="va">K</span>, q<span class="op">=</span><span class="fl">15</span>,  platform<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Visium'</span>, <span class="st">"ST"</span>, <span class="st">"Other_SRT"</span>, <span class="st">"scRNAseq"</span><span class="op">)</span>,<span class="va">...</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>seu</dt>
<dd><p>an object of class "Seurat". The details of this object are given under 'Details'.</p></dd>

  <dt>q</dt>
<dd><p>a positive integer, specify the number of latent features to be extracted, default as 15.</p></dd>

  <dt>K</dt>
<dd><p>a positive  integer or integer vector, specify the number of clusters. When <code>K</code> is a vector, it is automatically selected by MBIC criteria. Users can also use BIC and AIC  to select K or refine model with MBIC by argument <code>pen.const</code> in the function <a href="selectModel.html">selectModel</a>.</p></dd>

  <dt>platform</dt>
<dd><p>a string, specify the platform of the provided data, default as "Visium". There are more platforms to be chosen, including ("Visuim", "ST", "Other_SRT") and "scRNAseq", where the first group means there are spatial coordinates information in the metadata of seu, named "row" and "col" and a Hidden Markov random field is used to model the unobserved class label using spatial coordinates ("Other_SRT" represents the other SRT platforms except for 'Visium' and 'ST'), and the other group "scRNAseq" means there is no spatial information in object seu and a multinomial model is used to model the unobserved class labels. The platform helps to calculate the adjacency matrix.</p></dd>

  <dt>...</dt>
<dd><p>Other arguments to pass into <a href="DR.SC.fit.html">DR.SC_fit</a> function.</p></dd>

</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>seu is an object named <code>Seurat</code>, thich can easily created by R package <a href="https://satijalab.org/seurat/reference/Seurat-package.html" class="external-link">Seurat</a>.
DR-SC  model can be applied to analyze both single cell RNA sequencing data and spatially resoved transcriptomics (SRT) data.</p>
<p>If the data are collected by the single cell RNA sequencing techonologies which means there is no spatial information in object seu then a multinomial model is used to model the unobserved class labels.</p>

<p>If the data is collected by the spatial transcriptomics technologies, then there are spatial coordinates information in the metadata of seu, named "row" and "col". DR-SC model uses a Hidden Markov random field to model the spatial coordinates.  <code>DR.SC</code> supports different platforms of SRT data, such as `Visium`, `ST` and any other platforms `Other_SRT`.</p>
<p>For lattice grids in ST platform (ST), the interior spot has four neighbors (left, right, up and down),the boundary spot has three neighbors, and the spot in the corner has two neighbors. For hexagon grids, such as spatial coordinate in 10X Visium platform (Visium), the interior spot has six neighbors. For the irregular coordinates in other platforms (Other_SRT), Euclidean distance is adopted to decide whether a spot is an neighbor of another spot. For example, if the Euclidean distance between spot A and B is less than a radius, then A is taken as the neighbourhood of B. See function <a href="getAdj.html">getAdj</a> for more details.</p>
    </div>
    <div id="value">
    <h2>Value</h2>
    

<p>DR.SC returns a revised  <code>Seurat</code> object. There are two revisions in the seu: 1. the metadata is added a new column named <code>spatial.drsc.cluster</code> that represents the clustering results from DR-SC model, and the <code>Idents(seu)</code> is assigned with <code>spatial.drsc.cluster</code>. 2. a DimReduc object named <code>dr-sc</code> is added in the slot <code>reductions</code>, which represents the features extracted by DR-SC model.</p>


    </div>
    <div id="references">
    <h2>References</h2>
    <p><a href="https://academic.oup.com/nar/article/50/12/e72/6555431" class="external-link">Wei Liu, Xu Liao, Yi Yang, Huazhen Lin, Joe Yeong, Xiang Zhou, Xingjie Shi &amp; Jin Liu (2022). Joint dimension reduction and clustering analysis of single-cell RNA-seq and spatial transcriptomics data, Nucleic Acids Research, gkac219.</a></p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Wei Liu</p>
    </div>
    <div id="note">
    <h2>Note</h2>
    <p>nothing</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>None</p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="co">## we generate the spatial transcriptomics data with lattice neighborhood, i.e. ST platform.</span></span>
<span class="r-in"><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="genspatialdata.html">gendata_RNAExp</a></span><span class="op">(</span>height<span class="op">=</span><span class="fl">10</span>, width<span class="op">=</span><span class="fl">10</span>,p<span class="op">=</span><span class="fl">50</span>, K<span class="op">=</span><span class="fl">4</span>,platform<span class="op">=</span><span class="st">"ST"</span><span class="op">)</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span class="r-in"><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">seu</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-in"><span class="co"># choose 100 highly variable features</span></span>
<span class="r-in"><span class="co"># seu &lt;- FindVariableFeatures(seu, nfeatures = 100)</span></span>
<span class="r-in"><span class="co"># maxIter = 2 is only used for illustration, and user can use default.</span></span>
<span class="r-in"><span class="co"># seu1 &lt;- DR.SC(seu, K=4, platform = 'ST', maxIter=2,verbose=FALSE)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># choose spatially variable features (SVGs)</span></span>
<span class="r-in"><span class="va">seu</span> <span class="op">&lt;-</span> <span class="fu"><a href="FindSVGs.html">FindSVGs</a></span><span class="op">(</span><span class="va">seu</span>, nfeatures <span class="op">=</span> <span class="fl">40</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-in"><span class="co"># use SVGs to fit DR.SC model</span></span>
<span class="r-in"><span class="co"># maxIter = 2 is only used for illustration, and user can use default.</span></span>
<span class="r-in"><span class="va">seu1</span> <span class="op">&lt;-</span> <span class="fu">DR.SC</span><span class="op">(</span><span class="va">seu</span>, K<span class="op">=</span><span class="fl">4</span>,platform <span class="op">=</span> <span class="st">'ST'</span>, maxIter<span class="op">=</span><span class="fl">2</span>, verbose<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Neighbors were identified for 100 out of 100 spots.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Fit DR-SC model...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -------------------Calculate inital values-------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using accurate PCA to obtain initial values</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -------------------Finish computing inital values------------- </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -------------------Starting  ICM-EM algortihm-------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> iter = 2, loglik= -1167.507473, dloglik=0.999999 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -------------------Complete!-------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> elasped time is :0.05</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finish DR-SC model fitting</span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Wei Liu, Yi Yang, Jin Liu.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

