<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Joint dimension reduction and spatial clustering — DR.SC • DR.SC</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Joint dimension reduction and spatial clustering — DR.SC" />
<meta property="og:description" content="Joint dimension reduction and spatial clustering for scRNA-seq and spatial transcriptomics data" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DR.SC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/DR.SC.HCC1.html">DR-SC: HCC1 Data Analysis</a>
    </li>
    <li>
      <a href="../articles/DR.SC.Simu.html">DR-SC: installation and simulation</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Joint dimension reduction and spatial clustering</h1>
    
    <div class="hidden name"><code>DR.SC.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Joint dimension reduction and spatial clustering for scRNA-seq and spatial transcriptomics data</p>
    </div>

    <pre class="usage">DR.SC(object, ...)
### S3 method for class "Seurat"
DR.SC(seu, q=15, K=NULL, platform= c("Visium", "ST", 'scRNAseq'),
       nfeatures=2000,K_set = seq(2, 10), variable.type=c("HVGs", "SVGs") ...)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>seu</th>
      <td><p>an object of class "Seurat". The details of this object are given under 'Details'.</p></td>
    </tr>
    <tr>
      <th>q</th>
      <td><p>a positive integer, specify the number of latent features to be extracted, default as 15.</p></td>
    </tr>
    <tr>
      <th>K</th>
      <td><p>a positive  integer, specify the number of clusters, default as <code>NULL</code>. When <code>K=NULL</code>, it is automatically selected by MBIC criteria.</p></td>
    </tr>
    <tr>
      <th>platform</th>
      <td><p>a string, specify the platform of the provided data, default as "Visium". There are three platforms to be chosen, including "Visuim", "ST" and "scRNAseq", where the first two means there are spatial coordinates information in the metadata of seu, named "row" and "col" and a Hidden Markov random field is used to model the spatial coordinates, and the last one "scRNAseq" means there is no spatial information in object seu and a multinomial model is used to model the unobserved class labels. The platform helps to calculate the adjoint matrix.</p></td>
    </tr>
    <tr>
      <th>nfeatures</th>
      <td><p>a positive integer, means how many highly variable or spatially variable genes used for DR-SC model, default as 2000. If there are less than 2000 features in seu, then all features are used for DR-SC model.</p></td>
    </tr>
    <tr>
      <th>K_set</th>
      <td><p>an optional vector of positive integer, means the candidates of number of clusters used for MBIC.</p></td>
    </tr>
    <tr>
      <th>variable.type</th>
      <td><p>an optional string, specify whether use highly variable genes (HVGs) or spatially variable genes (SVGs) to fit DR-SC model, default as "HVGs".</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Other arguments to pass into <a href='DR.SC.fit.html'>DR.SC.fit</a> function.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>seu is an object named <code>Seurat</code>, thich can easily created by R package <a href='https://satijalab.org/seurat/reference/Seurat-package.html'>Seurat</a>.
  If the data is collected by the spatial transcriptomics technologies such as 10X Visium, ST, seqFISH, MERFISH and  Slide-seq, then there are spatial coordinates information in the metadata of seu, named "row" and "col". DR-SC model uses a Hidden Markov random field to model the spatial coordinates.  If the data is collected by the single cell RNA sequencing techonologies which means there is no spatial information in object seu then a multinomial model is used to model the unobserved class labels.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>DR.SC returns a revised  <code>Seurat</code> object. There are two revisions in the seu. 1. the metadata is added a new column named <code>spatial.drsc.cluster</code> that represents the clustering results from DR-SC model. 2. a DimReduc object named <code>dr-sc</code> is added in the slot <code>reductions</code>, which represents the features extracted by DR-SC model.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    
  

    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Wei Liu</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>nothing</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'>

</div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## we generate the spatial transcriptomics data with lattice neighborhood, i.e. ST platform.</span>
<span class='va'>seu</span> <span class='op'>&lt;-</span> <span class='fu'><a href='gendata_RNAExp.html'>gendata_RNAExp</a></span><span class='op'>(</span>height<span class='op'>=</span><span class='fl'>30</span>, width<span class='op'>=</span><span class='fl'>30</span>,p<span class='op'>=</span><span class='fl'>500</span>, K<span class='op'>=</span><span class='fl'>4</span>,platform<span class='op'>=</span><span class='st'>"ST"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='warning'>Warning: The following arguments are not used: row.names</span></div><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://satijalab.org/seurat'>Seurat</a></span><span class='op'>)</span>
<span class='va'>seu</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://satijalab.org/seurat/reference/NormalizeData.html'>NormalizeData</a></span><span class='op'>(</span><span class='va'>seu</span><span class='op'>)</span>
<span class='co'># choose 400 spatially variable features</span>
<span class='va'>seu</span> <span class='op'>&lt;-</span> <span class='fu'><a href='FindSVGs.html'>FindSVGs</a></span><span class='op'>(</span><span class='va'>seu</span>, nfeatures <span class='op'>=</span> <span class='fl'>400</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Loading required package: SPARK</span></div><div class='output co'>#&gt; Find the spatially variables genes by SPARK-X...
#&gt; ## ===== SPARK-X INPUT INFORMATION ==== 
#&gt; ## number of total samples: 900 
#&gt; ## number of total genes: 500 
#&gt; ## Running with single core, may take some time 
#&gt; ## Testing With Projection Kernel
#&gt; ## Testing With Gaussian Kernel 1
#&gt; ## Testing With Gaussian Kernel 2
#&gt; ## Testing With Gaussian Kernel 3
#&gt; ## Testing With Gaussian Kernel 4
#&gt; ## Testing With Gaussian Kernel 5
#&gt; ## Testing With Cosine Kernel 1
#&gt; ## Testing With Cosine Kernel 2
#&gt; ## Testing With Cosine Kernel 3
#&gt; ## Testing With Cosine Kernel 4
#&gt; ## Testing With Cosine Kernel 5</div><div class='input'><span class='va'>seu1</span> <span class='op'>&lt;-</span> <span class='fu'>DR.SC</span><span class='op'>(</span><span class='va'>seu</span>, K<span class='op'>=</span><span class='fl'>4</span>, platform <span class='op'>=</span> <span class='st'>'ST'</span>, verbose<span class='op'>=</span><span class='cn'>F</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Neighbors were identified for 900 out of 900 spots.</span></div><div class='output co'>#&gt; <span class='warning'>Warning: nrow(seu) is less than nfeatures, so assign nfeatures with nrow(seu)!</span></div><div class='output co'>#&gt; <span class='warning'>Warning: number of variable genes is less than nfeatures, </span>
#&gt; <span class='warning'>              so assign nfeatures with number of variable genes!</span></div><div class='output co'>#&gt; <span class='error'>Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 't': 'NA' indices are not (yet?) supported for sparse Matrices</span></div><div class='input'>
<span class='co'># choose spatially variable features</span>
<span class='va'>seu</span> <span class='op'>&lt;-</span> <span class='fu'><a href='FindSVGs.html'>FindSVGs</a></span><span class='op'>(</span><span class='va'>seu</span><span class='op'>)</span>
</div><div class='output co'>#&gt; Find the spatially variables genes by SPARK-X...
#&gt; ## ===== SPARK-X INPUT INFORMATION ==== 
#&gt; ## number of total samples: 900 
#&gt; ## number of total genes: 500 
#&gt; ## Running with single core, may take some time 
#&gt; ## Testing With Projection Kernel
#&gt; ## Testing With Gaussian Kernel 1
#&gt; ## Testing With Gaussian Kernel 2
#&gt; ## Testing With Gaussian Kernel 3
#&gt; ## Testing With Gaussian Kernel 4
#&gt; ## Testing With Gaussian Kernel 5
#&gt; ## Testing With Cosine Kernel 1
#&gt; ## Testing With Cosine Kernel 2
#&gt; ## Testing With Cosine Kernel 3
#&gt; ## Testing With Cosine Kernel 4
#&gt; ## Testing With Cosine Kernel 5</div><div class='input'><span class='co'># use SVGs to fit DR.SC model</span>
<span class='va'>seu1</span> <span class='op'>&lt;-</span> <span class='fu'>DR.SC</span><span class='op'>(</span><span class='va'>seu</span>, K<span class='op'>=</span><span class='fl'>4</span>,platform <span class='op'>=</span> <span class='st'>'ST'</span>, variable.type <span class='op'>=</span> <span class='st'>'SVGs'</span>,verbose<span class='op'>=</span><span class='cn'>F</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Neighbors were identified for 900 out of 900 spots.</span></div><div class='output co'>#&gt; <span class='warning'>Warning: nrow(seu) is less than nfeatures, so assign nfeatures with nrow(seu)!</span></div><div class='output co'>#&gt; Using SVGs to fit DR.SC model since variable.type=SVGs...
#&gt; Fit DR-SC model...
#&gt; elasped time is : 15.18 1.09 16.22 NA NA 
#&gt; Finish DR-SC model fitting</div><div class='input'>
<span class='co'>## genrate the scRNAseq data</span>
<span class='va'>seu</span> <span class='op'>&lt;-</span> <span class='fu'><a href='gendata_RNAExp.html'>gendata_RNAExp</a></span><span class='op'>(</span>height<span class='op'>=</span><span class='fl'>30</span>, width<span class='op'>=</span><span class='fl'>30</span>,p<span class='op'>=</span><span class='fl'>500</span>, K<span class='op'>=</span><span class='fl'>4</span>,platform<span class='op'>=</span><span class='st'>"scRNAseq"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='warning'>Warning: The following arguments are not used: row.names</span></div><div class='input'><span class='va'>seu</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://satijalab.org/seurat/reference/NormalizeData.html'>NormalizeData</a></span><span class='op'>(</span><span class='va'>seu</span><span class='op'>)</span>
<span class='co'># choose 400 variable features using Seurat</span>
<span class='va'>seu</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://satijalab.org/seurat/reference/FindVariableFeatures.html'>FindVariableFeatures</a></span><span class='op'>(</span><span class='va'>seu</span>, nfeatures <span class='op'>=</span> <span class='fl'>400</span><span class='op'>)</span>
<span class='va'>seu1</span> <span class='op'>&lt;-</span> <span class='fu'>DR.SC</span><span class='op'>(</span><span class='va'>seu</span>, K<span class='op'>=</span><span class='fl'>4</span>, platform <span class='op'>=</span> <span class='st'>'scRNAseq'</span>,nfeatures<span class='op'>=</span><span class='fl'>400</span>, verbose<span class='op'>=</span><span class='cn'>F</span><span class='op'>)</span>
</div><div class='output co'>#&gt; Fit DR-SC model...
#&gt; elasped time is : 10.3 0.82 11.25 NA NA 
#&gt; Finish DR-SC model fitting</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Wei Liu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


