<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Joint dimension reduction and spatial clustering — DR.SC.fit • DR.SC</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Joint dimension reduction and spatial clustering — DR.SC.fit" />
<meta property="og:description" content="Joint dimension reduction and spatial clustering for scRNA-seq and spatial transcriptomics data" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DR.SC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/DR.SC.HCC1.html">DR-SC: HCC1 Data Analysis</a>
    </li>
    <li>
      <a href="../articles/DR.SC.Simu.html">DR-SC: installation and simulation</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Joint dimension reduction and spatial clustering</h1>
    
    <div class="hidden name"><code>DR.SC.fit.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Joint dimension reduction and spatial clustering for scRNA-seq and spatial transcriptomics data</p>
    </div>

    <pre class="usage"><span class='fu'>DR.SC.fit</span><span class='op'>(</span><span class='va'>X</span>,Adj_sp<span class='op'>=</span><span class='cn'>NULL</span>, q<span class='op'>=</span><span class='fl'>15</span>, K<span class='op'>=</span> <span class='cn'>NULL</span>,
        error.heter<span class='op'>=</span> <span class='cn'>T</span>, K_set <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>2</span>, <span class='fl'>10</span><span class='op'>)</span>, beta_grid<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>0.5</span>, <span class='fl'>5</span>, by<span class='op'>=</span><span class='fl'>0.5</span><span class='op'>)</span>,
        maxIter<span class='op'>=</span><span class='fl'>30</span>, epsLogLik<span class='op'>=</span><span class='fl'>1e-5</span>, verbose<span class='op'>=</span><span class='cn'>F</span>, maxIter_ICM<span class='op'>=</span><span class='fl'>6</span>,pen.const<span class='op'>=</span><span class='fl'>1</span>,
        wpca.int<span class='op'>=</span><span class='cn'>F</span>, parallel<span class='op'>=</span><span class='st'>'doSNOW'</span>, num_core<span class='op'>=</span><span class='fl'>5</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>X</th>
      <td><p>an optional sparse matrix with class <code>dgCMatrix</code>, specify the log-normalization gene expression matrix used for DR-SC model.</p></td>
    </tr>
    <tr>
      <th>Adj_sp</th>
      <td><p>an optional sparse matrix with class <code>dgCMatrix</code>, specify the adjoint matrix used for DR-SC model. We provide this interface for those users  who would like to define the adjoint matrix by their own.</p></td>
    </tr>
    <tr>
      <th>q</th>
      <td><p>a positive integer, specify the number of latent features to be extracted, default as 15.</p></td>
    </tr>
    <tr>
      <th>K</th>
      <td><p>a positive  integer, specify the number of clusters, default as <code>NULL</code>. When <code>K=NULL</code>, it is automatically selected by MBIC criteria.</p></td>
    </tr>
    <tr>
      <th>K_set</th>
      <td><p>a vector of positive integer, means the candidates of number of clusters used for MBIC.</p></td>
    </tr>
    <tr>
      <th>error.heter</th>
      <td><p>an optional logical value, whether use the heterogenous error for DR-SC model, default as <code>TRUE</code>. If <code>error.heter=FALSE</code>, then the homogenuous error is used for model.</p></td>
    </tr>
    <tr>
      <th>beta_grid</th>
      <td><p>an optional vector of positive value, the candidate set of the smoothing parameter to be searched by the grid-search optimization approach.</p></td>
    </tr>
    <tr>
      <th>maxIter</th>
      <td><p>an optional positive value, represents the maximum iterations of EM.</p></td>
    </tr>
    <tr>
      <th>epsLogLik</th>
      <td><p>an optional positive vlaue, tolerance vlaue of relative variation rate of the observed pseudo log-loglikelihood value, defualt as '1e-5'.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>an optional logical value, whether output the information of the ICM-EM algorithm.</p></td>
    </tr>
    <tr>
      <th>maxIter_ICM</th>
      <td><p>an optional positive value, represents the maximum iterations of ICM.</p></td>
    </tr>
    <tr>
      <th>pen.const</th>
      <td><p>an optional positive value, the adjusted constant used in the MBIC criteria. It usually takes value between 0.1 to 1.</p></td>
    </tr>
    <tr>
      <th>wpca.int</th>
      <td><p>an optional logical value, means whether use the weighted PCA to obtain the initial values of loadings and other paramters, default as <code>FALSE</code> which means the conventional PCA  is used.</p></td>
    </tr>
    <tr>
      <th>parallel</th>
      <td><p>a optional string, specify the parallel way to choose the number of clusters by MBIC. We provide three methods: 1. <code>parallel="doSNOW"</code> uses <code>doSNOW</code> R package to conduct the parallel schema; 2. <code>parallel="parrallel"</code> uses <code>parrallel</code> R package to conduct the parallel schema; 3.<code>parallel=NULL</code> doesn't use parallel computation.</p></td>
    </tr>
    <tr>
      <th>num_core</th>
      <td><p>an optional positive integer, means the cores used in parallel computating.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Nothing</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>DR.SC.fit returns a <a href='https://rdrr.io/r/base/list.html'>list</a> with the following components:</p>
<dt>cluster</dt><dd><p>inferred class labels</p></dd>
<dt>hZ</dt><dd><p>extracted latent features.</p></dd>
<dt>beta</dt><dd><p>estimated smoothing parameter</p></dd>
<dt>Mu</dt><dd><p>mean vectors of mixtures components.</p></dd>
<dt>Sigma</dt><dd><p>covariance matrix of mixtures components.</p></dd>
<dt>W</dt><dd><p>estimated loading matrix</p></dd>
<dt>Lam_vec</dt><dd><p>estimated variance of errors in probabilistic PCA model</p></dd>
<dt>loglik</dt><dd><p>pseudo observed log-likelihood</p></dd>

    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    
  

    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Wei Liu</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>nothing</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'>

</div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## we generate the spatial transcriptomics data with lattice neighborhood, i.e. ST platform.</span>
<span class='va'>seu</span> <span class='op'>&lt;-</span> <span class='fu'><a href='gendata_RNAExp.html'>gendata_RNAExp</a></span><span class='op'>(</span>height<span class='op'>=</span><span class='fl'>30</span>, width<span class='op'>=</span><span class='fl'>30</span>,p<span class='op'>=</span><span class='fl'>500</span>, K<span class='op'>=</span><span class='fl'>4</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Loading required package: GiRaF</span></div><div class='output co'>#&gt; <span class='warning'>Warning: package 'GiRaF' was built under R version 4.0.4</span></div><div class='output co'>#&gt; <span class='warning'>Warning: The following arguments are not used: row.names</span></div><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://satijalab.org/seurat'>Seurat</a></span><span class='op'>)</span>
<span class='va'>seu</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://satijalab.org/seurat/reference/NormalizeData.html'>NormalizeData</a></span><span class='op'>(</span><span class='va'>seu</span><span class='op'>)</span>
<span class='co'># choose 2000 variable features using Seurat</span>
<span class='va'>seu</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://satijalab.org/seurat/reference/FindVariableFeatures.html'>FindVariableFeatures</a></span><span class='op'>(</span><span class='va'>seu</span>, nfeatures <span class='op'>=</span> <span class='fl'>2000</span><span class='op'>)</span>
<span class='co'># users define the adjoint matrix</span>
<span class='va'>Adj_sp</span> <span class='op'>&lt;-</span> <span class='fu'><a href='getAdj.html'>getAdj</a></span><span class='op'>(</span><span class='va'>seu</span>, platform <span class='op'>=</span> <span class='st'>'ST'</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Neighbors were identified for 900 out of 900 spots.</span></div><div class='input'><span class='va'>var.features</span> <span class='op'>&lt;-</span> <span class='va'>seu</span><span class='op'>@</span><span class='va'>assays</span><span class='op'>$</span><span class='va'>RNA</span><span class='op'>@</span><span class='va'>var.features</span>
<span class='va'>X</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='fu'><a href='https://satijalab.org/seurat/reference/LogNormalize.html'>LogNormalize</a></span><span class='op'>(</span><span class='va'>seu</span><span class='op'>@</span><span class='va'>assays</span><span class='op'>$</span><span class='va'>RNA</span><span class='op'>@</span><span class='va'>counts</span><span class='op'>[</span><span class='va'>var.features</span>,<span class='op'>]</span>,<span class='op'>)</span><span class='op'>)</span>
<span class='va'>drscList</span> <span class='op'>&lt;-</span> <span class='fu'>DR.SC.fit</span><span class='op'>(</span><span class='va'>X</span>,Adj_sp<span class='op'>=</span><span class='va'>Adj_sp</span> ,K<span class='op'>=</span><span class='fl'>4</span>,  verbose<span class='op'>=</span><span class='cn'>T</span><span class='op'>)</span>
</div><div class='output co'>#&gt; Fit DR-SC model...
#&gt; -------------------Calculate inital values------------- 
#&gt; -------------------Finish computing inital values------------- 
#&gt; -------------------Starting  EM algortihm------------- 
#&gt; -------------------Complete!------------- 
#&gt; elasped time is : 16.28 1.05 17.24 NA NA 
#&gt; Finish DR-SC model fitting</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Wei Liu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


